
<!-- rnb-text-begin -->

---
title: "BCB420 Assignment #1: Data set selection and initial Processing"
output:
  html_notebook:
    toc: yes
bibliography: references.bib
---


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImxpYnJhcnkoXCJHRU9xdWVyeVwiKSIsImxpYnJhcnkoXCJrbml0clwiKSIsImxpYnJhcnkoXCJlZGdlUlwiKSIsImxpYnJhcnkoXCJiaW9tYVJ0XCIpIiwibGlicmFyeShcInN0cmluZ3JcIikiLCJsaWJyYXJ5KFwibGltbWFcIikiXX0= -->

```r
library("GEOquery")
library("knitr")
library("edgeR")
library("biomaRt")
library("stringr")
library("limma")
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


# BCB420 Assignment #1: Data Set Selection and Initial Processing

By: Yunyang Harrison Deng

## Data set's Importance

This data set is of particular interest to me since I have previously worked on multiple projects on the topic of HIV. However, up until this point, I have not gotten the opportunity to perform computational analysis with expression data sets so reading the paper, working with the data is all very novel and interesting to me. Furthermore, the data set was produced for a study regarding elite controllers, those that have previously been exposed to HIV, but is able to maintain an undetectable viral load without any form of treatment for years. This discussion for elite controllers is especially interesting for me since I have ongoing projects looking into natural resistance to HIV and immune escape mechanisms on the genetics level. However, expression data could be beneficial in not only corroboration, but isolation in which factors to look at at the genetics level.

## Variables of Data Set

The data set contains various conditions to stratify and categorize data based on. Importantly, the following analysis will focus on the phenotype category which contains antiretroviral therapy (ART) patient sample, elite controller patient sample where no plasma viremia was detected (EC), or viremic control where very low levels of viremia was detected (VC). For the purposes of this analysis, we will use ART patient samples as the control to see if there are differences between how spontaneous controllers react to HIV versus the artificial treatment patients react [@collinsCytolyticCD8Cells2023].

## Loading the Data

We will begin by downloading the data with use of the GEOquery library.

### Fetching GEO Data


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImRhdGFfc2V0X2dlb2lkIDwtIFwiR1NFMTk2NTQ5XCIiLCJnc2UgPC0gZ2V0R0VPKGRhdGFfc2V0X2dlb2lkICxHU0VNYXRyaXg9RkFMU0UpICMgRmV0Y2hlcyBHRU8gU09GVCBhbmQgcGFyc2VzLiIsImdzZUBoZWFkZXIkc3VtbWFyeSJdfQ== -->

```r
data_set_geoid <- "GSE196549"
gse <- getGEO(data_set_geoid ,GSEMatrix=FALSE) # Fetches GEO SOFT and parses.
gse@header$summary
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIFwiRm9sbGljdWxhciBDRDgrIFQgY2VsbHMgKGZDRDgpIG1lZGlhdGUgc3VydmVpbGxhbmNlIGluIGx5bXBoIG5vZGUgKExOKSBnZXJtaW5hbCBjZW50ZXJzIGFnYWluc3QgbHltcGhvdHJvcGljIGluZmVjdGlvbnMgYW5kIGNhbmNlcnMsIGJ1dCBwcmVjaXNlIG1lY2hhbmlzbXMgYnkgd2hpY2ggdGhlc2UgY2VsbHMgbWVkaWF0ZSBpbW11bmUgY29udHJvbCByZW1haW4gaW5jb21wbGV0ZWx5IHJlc29sdmVkLiBUbyBhZGRyZXNzIHRoaXMsIHdlIGludmVzdGlnYXRlZCBmdW5jdGlvbmFsaXR5LCBjbG9ub3R5cGljIGNvbXBhcnRtZW50YWxpemF0aW9uLCBzcGF0aWFsIGxvY2FsaXphdGlvbiwgcGhlbm90eXBpYyBjaGFyYWN0ZXJpc3RpY3MsIGFuZCB0cmFuc2NyaXB0aW9uYWwgcHJvZmlsZXMgb2YgTE4tcmVzaWRlbnQgdmlydXMtc3BlY2lmaWMgQ0Q4KyBUIGNlbGxzIGluIHBlcnNvbnMgd2hvIGNvbnRyb2wgSElWIHdpdGhvdXQgbWVkaWNhdGlvbnMuIEFudGlnZW4taW5kdWNlZCBwcm9saWZlcmF0aXZlIGFuZCBjeXRvbHl0aWMgcG90ZW50aWFsIGNvbnNpc3RlbnRseSBkaXN0aW5ndWlzaGVkIHNwb250YW5lb3VzIGNvbnRyb2xsZXJzIGZyb20gbm9uY29udHJvbGxlcnMuIFQgY2VsbCByZWNlcHRvciBhbmFseXNpcyByZXZlYWxlZCBjb21wbGV0ZSBjbG9ub3R5cGljIG92ZXJsYXAgYmV0d2VlbiBwZXJpcGhlcmFsIGFuZCBMTi1yZXNpZGVudCBISVYtc3BlY2lmaWMgQ0Q4KyBUIGNlbGxzLiBUcmFuc2NyaXB0aW9uYWwgYW5hbHlzaXMgb2YgTE4gQ0Q4KyBUIGNlbGxzIHJldmVhbGVkIGdlbmUgc2lnbmF0dXJlcyBvZiBpbmZsYW1tYXRvcnkgY2hlbW90YXhpcyBhbmQgYW50aWdlbi1pbmR1Y2VkIGVmZmVjdG9yIGZ1bmN0aW9uLiBJbiBISVYgY29udHJvbGxlcnMsIHRoZSBjeXRvdG94aWMgZWZmZWN0b3JzIHBlcmZvcmluIGFuZCBncmFuenltZSBCIHdlcmUgZWxldmF0ZWQgYW1vbmcgdmlydXMtc3BlY2lmaWMgQ1hDUjUrIGZDRDhzIHByb3hpbWF0ZSB0byBmb2NpIG9mIEhJViBSTkEgd2l0aGluIGdlcm1pbmFsIGNlbnRlcnMuIFRoZXNlIHJlc3VsdHMgcHJvdmlkZSBldmlkZW5jZSBjb25zaXN0ZW50IHdpdGggY3l0b2x5dGljIGNvbnRyb2wgb2YgbHltcGhvdHJvcGljIGluZmVjdGlvbiBzdXBwb3J0ZWQgYnkgaW5mbGFtbWF0b3J5IHJlY3J1aXRtZW50LCBhbnRpZ2VuLXNwZWNpZmljIHByb2xpZmVyYXRpb24sIGFuZCBjeXRvdG94aWNpdHkgb2YgZkNEOHMuXCJcbiJ9 -->

```
[1] "Follicular CD8+ T cells (fCD8) mediate surveillance in lymph node (LN) germinal centers against lymphotropic infections and cancers, but precise mechanisms by which these cells mediate immune control remain incompletely resolved. To address this, we investigated functionality, clonotypic compartmentalization, spatial localization, phenotypic characteristics, and transcriptional profiles of LN-resident virus-specific CD8+ T cells in persons who control HIV without medications. Antigen-induced proliferative and cytolytic potential consistently distinguished spontaneous controllers from noncontrollers. T cell receptor analysis revealed complete clonotypic overlap between peripheral and LN-resident HIV-specific CD8+ T cells. Transcriptional analysis of LN CD8+ T cells revealed gene signatures of inflammatory chemotaxis and antigen-induced effector function. In HIV controllers, the cytotoxic effectors perforin and granzyme B were elevated among virus-specific CXCR5+ fCD8s proximate to foci of HIV RNA within germinal centers. These results provide evidence consistent with cytolytic control of lymphotropic infection supported by inflammatory recruitment, antigen-specific proliferation, and cytotoxicity of fCD8s."
```



<!-- rnb-output-end -->

<!-- rnb-output-begin eyJkYXRhIjoiRm9sbGljdWxhciBDRDgrIFQgY2VsbHMgKGZDRDgpIG1lZGlhdGUgc3VydmVpbGxhbmNlIGluIGx5bXBoIG5vZGUgKExOKSBnZXJtaW5hbCBjZW50ZXJzIGFnYWluc3QgbHltcGhvdHJvcGljIGluZmVjdGlvbnMgYW5kIGNhbmNlcnMsIGJ1dCBwcmVjaXNlIG1lY2hhbmlzbXMgYnkgd2hpY2ggdGhlc2UgY2VsbHMgbWVkaWF0ZSBpbW11bmUgY29udHJvbCByZW1haW4gaW5jb21wbGV0ZWx5IHJlc29sdmVkLiBUbyBhZGRyZXNzIHRoaXMsIHdlIGludmVzdGlnYXRlZCBmdW5jdGlvbmFsaXR5LCBjbG9ub3R5cGljIGNvbXBhcnRtZW50YWxpemF0aW9uLCBzcGF0aWFsIGxvY2FsaXphdGlvbiwgcGhlbm90eXBpYyBjaGFyYWN0ZXJpc3RpY3MsIGFuZCB0cmFuc2NyaXB0aW9uYWwgcHJvZmlsZXMgb2YgTE4tcmVzaWRlbnQgdmlydXMtc3BlY2lmaWMgQ0Q4KyBUIGNlbGxzIGluIHBlcnNvbnMgd2hvIGNvbnRyb2wgSElWIHdpdGhvdXQgbWVkaWNhdGlvbnMuIEFudGlnZW4taW5kdWNlZCBwcm9saWZlcmF0aXZlIGFuZCBjeXRvbHl0aWMgcG90ZW50aWFsIGNvbnNpc3RlbnRseSBkaXN0aW5ndWlzaGVkIHNwb250YW5lb3VzIGNvbnRyb2xsZXJzIGZyb20gbm9uY29udHJvbGxlcnMuIFQgY2VsbCByZWNlcHRvciBhbmFseXNpcyByZXZlYWxlZCBjb21wbGV0ZSBjbG9ub3R5cGljIG92ZXJsYXAgYmV0d2VlbiBwZXJpcGhlcmFsIGFuZCBMTi1yZXNpZGVudCBISVYtc3BlY2lmaWMgQ0Q4KyBUIGNlbGxzLiBUcmFuc2NyaXB0aW9uYWwgYW5hbHlzaXMgb2YgTE4gQ0Q4KyBUIGNlbGxzIHJldmVhbGVkIGdlbmUgc2lnbmF0dXJlcyBvZiBpbmZsYW1tYXRvcnkgY2hlbW90YXhpcyBhbmQgYW50aWdlbi1pbmR1Y2VkIGVmZmVjdG9yIGZ1bmN0aW9uLiBJbiBISVYgY29udHJvbGxlcnMsIHRoZSBjeXRvdG94aWMgZWZmZWN0b3JzIHBlcmZvcmluIGFuZCBncmFuenltZSBCIHdlcmUgZWxldmF0ZWQgYW1vbmcgdmlydXMtc3BlY2lmaWMgQ1hDUjUrIGZDRDhzIHByb3hpbWF0ZSB0byBmb2NpIG9mIEhJViBSTkEgd2l0aGluIGdlcm1pbmFsIGNlbnRlcnMuIFRoZXNlIHJlc3VsdHMgcHJvdmlkZSBldmlkZW5jZSBjb25zaXN0ZW50IHdpdGggY3l0b2x5dGljIGNvbnRyb2wgb2YgbHltcGhvdHJvcGljIGluZmVjdGlvbiBzdXBwb3J0ZWQgYnkgaW5mbGFtbWF0b3J5IHJlY3J1aXRtZW50LCBhbnRpZ2VuLXNwZWNpZmljIHByb2xpZmVyYXRpb24sIGFuZCBjeXRvdG94aWNpdHkgb2YgZkNEOHMuXG4ifQ== -->

```
Follicular CD8+ T cells (fCD8) mediate surveillance in lymph node (LN) germinal centers against lymphotropic infections and cancers, but precise mechanisms by which these cells mediate immune control remain incompletely resolved. To address this, we investigated functionality, clonotypic compartmentalization, spatial localization, phenotypic characteristics, and transcriptional profiles of LN-resident virus-specific CD8+ T cells in persons who control HIV without medications. Antigen-induced proliferative and cytolytic potential consistently distinguished spontaneous controllers from noncontrollers. T cell receptor analysis revealed complete clonotypic overlap between peripheral and LN-resident HIV-specific CD8+ T cells. Transcriptional analysis of LN CD8+ T cells revealed gene signatures of inflammatory chemotaxis and antigen-induced effector function. In HIV controllers, the cytotoxic effectors perforin and granzyme B were elevated among virus-specific CXCR5+ fCD8s proximate to foci of HIV RNA within germinal centers. These results provide evidence consistent with cytolytic control of lymphotropic infection supported by inflammatory recruitment, antigen-specific proliferation, and cytotoxicity of fCD8s.
```



<!-- rnb-output-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Seeing as the data is downloaded correctly, we can check the information regarding the platform used to generate the data.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImN1cnJlbnRfZ3BsIDwtIG5hbWVzKEdQTExpc3QoZ3NlKSkiLCJjdXJyZW50X2dwbF9pbmZvIDwtIE1ldGEoZ2V0R0VPKGN1cnJlbnRfZ3BsKSkiXX0= -->

```r
current_gpl <- names(GPLList(gse))
current_gpl_info <- Meta(getGEO(current_gpl))
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


-   Platform title: NextSeq 550 (Homo sapiens)
-   Submission date: Apr 05 2016
-   Last update: Dec 05 2018
-   Organisms: Homo sapiens
-   Number of GEO datasets using this platform: 1232
-   Number of GEO samples that use this technology: 33612

Next, we'll take a quick look at the processing methodology information embedded in the individual sample records:


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjoiZm9yIChpbmZvIGluIGdzZUBnc21zW1sxXV1AaGVhZGVyJGRhdGFfcHJvY2Vzc2luZykge1xuICBwcmludChpbmZvKVxufSJ9 -->

```r
for (info in gse@gsms[[1]]@header$data_processing) {
  print(info)
}
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIFwiSWxsdW1pbmEgc2VxdWVuY2luZyBydW4gd2FzIGRlZW11bHRpcGxleGVkIHVzaW5nIGJjbDJmYXN0cSB2ZWVyc2lvbiAyLjE5LjEuNDAzLlwiXG5bMV0gXCJSZWFkcyB3ZXJlIHRoZW4gbWFwcGVlZCB0byB0aGUgaGczOCB0cmFuc2NyaXB0b21lIHVzaW5nIHRoZSBSU0VNIHBpcGVsaW5lIHZlcnNpb24gMS4zLjAgd2l0aCBib3d0aWUgYWxpZ25lciB2ZXJzaW9uIDEuMi4yLiBSZWFkIGNvdW50cyB3ZXJlIGNhbGN1bGF0ZWQgYXMgZXhwZWN0ZWQgY291bnRzLCBhbmQgcm91bmRlZCB0byBpbnRlZ2VyIHZhbHVlcyBmb3IgYW5hbHlzaXMgd2l0aCBERVNlcTIsIG9yIHVzZWQgd2l0aG91dCByb3VuZGluZyBpbiBlZGdlUi5cIlxuWzFdIFwiRGlmZmVyZW50aWFsIGV4cHJlc3Npb24gYW5hbHlzaXMgd2FzIHBlcmZvcm1lZCB1c2luZyBERVNlcTIgdmVyc2lvbiAxLjI2LjAgdXNpbmcgYSBtb2RlbCBpbmNsdWRpbmcgdGhlIGZvbGxvd2luZyB0ZXJtczogXFxcIn4gcGF0aWVudF9JRCArIHRpc3N1ZSArIEdlbmVzRXhwQ291bnRHUkUxICsgYWxpZ25fcmF0ZSArIHBjdEdDLlIxXFxcIlwiXG5bMV0gXCJQcmlvciB0byBkaWZmZXJlbnRpYWwgZXhwcmVzc2lvbiBhbmFseXNpcywgZ2VuZXMgd2VyZSBmaWx0ZXJlZCB0byBoYXZlIGF0IGxlYXN0IDUgZXhwZWN0ZWQgY291bnRzIGluIGF0IGxlYXN0IDEwJSBvZiBzYW1wbGVzLlwiXG5bMV0gXCJHZW5vbWVfYnVpbGQ6IEdSQ2gzOC5wNVwiXG5bMV0gXCJTdXBwbGVtZW50YXJ5X2ZpbGVzX2Zvcm1hdF9hbmRfY29udGVudDogVGhlIGV4cGVjdGVkIGNvdW50cyBtYXRyaXggKFIxMDE1UjEyMDAuRklMVC5FQy50c3YpIGluY2x1ZGVzIGNvdW50cyBmb3Igc2FtcGxlcyBhbmQgZ2VuZXMgbGlzdGVkIGJ5IHRoZWlyIHJlc3BlY3RpdmUgRU5TRU1CTCBnZW5lIElEcyBhbmQgZ2VuZSBzeW1ib2xzLCBkZWxpbWl0ZWQgYnkgYW4gdW5kZXJzY29yZS4gUjEwMTVSMTIwMC5GSUxULkVDLkxUTU0udHN2IGNvbnRhaW5zIHRoZSBjb3JyZXNwb25kaW5nIFRNTS1ub3JtYWxpemVkIGxvZyB0cmFuc2Zvcm1lZCBjb3VudHMuIFRoZSBHRU8ubWV0YWRhdGEgZmlsZSBpcyBjb21wbGV0ZSBzYW1wbGUgbWV0YWRhdGEgaW5jbHVkaW5nIFFDIGV2YWx1YXRpb25zIGluIHRhYi1zZXBhcmF0ZWQgZm9ybWF0LiBSMTAxNVIxMjAwLkxOdlBCLnJzbHQudHN2IGlzIHRoZSBERVNlcTIgcmVzdWx0IHNldCBhcyB0YWItc2VwYXJhdGVkIHZhbHVlcy5cIlxuIn0= -->

```
[1] "Illumina sequencing run was deemultiplexed using bcl2fastq veersion 2.19.1.403."
[1] "Reads were then mappeed to the hg38 transcriptome using the RSEM pipeline version 1.3.0 with bowtie aligner version 1.2.2. Read counts were calculated as expected counts, and rounded to integer values for analysis with DESeq2, or used without rounding in edgeR."
[1] "Differential expression analysis was performed using DESeq2 version 1.26.0 using a model including the following terms: \"~ patient_ID + tissue + GenesExpCountGRE1 + align_rate + pctGC.R1\""
[1] "Prior to differential expression analysis, genes were filtered to have at least 5 expected counts in at least 10% of samples."
[1] "Genome_build: GRCh38.p5"
[1] "Supplementary_files_format_and_content: The expected counts matrix (R1015R1200.FILT.EC.tsv) includes counts for samples and genes listed by their respective ENSEMBL gene IDs and gene symbols, delimited by an underscore. R1015R1200.FILT.EC.LTMM.tsv contains the corresponding TMM-normalized log transformed counts. The GEO.metadata file is complete sample metadata including QC evaluations in tab-separated format. R1015R1200.LNvPB.rslt.tsv is the DESeq2 result set as tab-separated values."
```



<!-- rnb-output-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


### Fetching Study Supplementary Files

Then, we'll download all discovered supplementary files again using GEOQuery [@davisGEOqueryBridgeGene2007].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbIiMgR2V0IGFsbCBrbm93biBzdXBwbGVtZW50YXJ5IGZpbGUgaW5mb3JtYXRpb24uIiwic2ZpbGVuYW1lcyA9IGdldEdFT1N1cHBGaWxlcyhkYXRhX3NldF9nZW9pZCwgZmV0Y2hfZmlsZXMgPSBGQUxTRSkiLCJzZmlsZW5hbWVzJGZuYW1lIl19 -->

```r
# Get all known supplementary file information.
sfilenames = getGEOSuppFiles(data_set_geoid, fetch_files = FALSE)
sfilenames$fname
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIFwiR1NFMTk2NTQ5X0dFTy5tZXRhZGF0YS50c3YuZ3pcIiAgICAgICAgICAgXG5bMl0gXCJHU0UxOTY1NDlfUjEwMTVSMTIwMC5GSUxULkVDLkxUTU0udHN2Lmd6XCJcblszXSBcIkdTRTE5NjU0OV9SMTAxNVIxMjAwLkZJTFQuRUMudHN2Lmd6XCIgICAgIFxuWzRdIFwiR1NFMTk2NTQ5X1IxMDE1UjEyMDAuTE52UEIucnNsdC50c3YuZ3pcIiAgXG4ifQ== -->

```
[1] "GSE196549_GEO.metadata.tsv.gz"           
[2] "GSE196549_R1015R1200.FILT.EC.LTMM.tsv.gz"
[3] "GSE196549_R1015R1200.FILT.EC.tsv.gz"     
[4] "GSE196549_R1015R1200.LNvPB.rslt.tsv.gz"  
```



<!-- rnb-output-end -->

<!-- rnb-output-begin eyJkYXRhIjpbIkdTRTE5NjU0OV9HRU8ubWV0YWRhdGEudHN2Lmd6XG4iLCJHU0UxOTY1NDlfUjEwMTVSMTIwMC5GSUxULkVDLkxUTU0udHN2Lmd6XG4iLCJHU0UxOTY1NDlfUjEwMTVSMTIwMC5GSUxULkVDLnRzdi5nelxuIiwiR1NFMTk2NTQ5X1IxMDE1UjEyMDAuTE52UEIucnNsdC50c3YuZ3pcbiJdfQ== -->

```
GSE196549_GEO.metadata.tsv.gz

GSE196549_R1015R1200.FILT.EC.LTMM.tsv.gz

GSE196549_R1015R1200.FILT.EC.tsv.gz

GSE196549_R1015R1200.LNvPB.rslt.tsv.gz
```



<!-- rnb-output-end -->

<!-- rnb-source-begin eyJkYXRhIjpbIiMgRG93bmxvYWQgbG9jYXRpb24iLCJkb3dubG9hZF9kaXIgPC0gZmlsZS5wYXRoKGZpbGUucGF0aChnZXR3ZCgpLCBcImV4cHJlc3Npb25fZGF0YVwiKSkiLCJpZiAoIWRpci5leGlzdHMoZG93bmxvYWRfZGlyKSkge1xuICBkaXIuY3JlYXRlKGRvd25sb2FkX2RpcikgIyBDcmVhdGUgZm9sZGVyIGlmIGl0IGRvZXNuJ3QgZXhpc3QuXG59IiwiIiwiIyBGaWd1cmUgb3V0IHdoaWNoIGZpbGVzIGFyZSBtaXNzaW5nIiwibWlzc2luZ19maWxlcyA8LSBzZmlsZW5hbWVzJGZuYW1lWyF1bmxpc3QoXG4gIGxhcHBseShcbiAgICBzZmlsZW5hbWVzJGZuYW1lLCBGVU49ZnVuY3Rpb24oeCl7XG4gICAgICBmaWxlLmV4aXN0cyhmaWxlLnBhdGgoZG93bmxvYWRfZGlyLGRhdGFfc2V0X2dlb2lkLHgpKVxuICAgIH1cbiAgKVxuKV0iLCIiLCJpZihsZW5ndGgobWlzc2luZ19maWxlcykgPjApe1xuICBmb3IoaSBpbiAxOmxlbmd0aChtaXNzaW5nX2ZpbGVzKSkge1xuICAgICNnZXQgdGhlIHN1cHBsZW1lbnRhcnkgZmlsZXNcbiAgICBzZmlsZXMgPSBnZXRHRU9TdXBwRmlsZXMoZGF0YV9zZXRfZ2VvaWQsXG4gICAgICBmaWx0ZXJfcmVnZXggPSBtaXNzaW5nX2ZpbGVzW2ldLFxuICAgICAgYmFzZURpciA9IGRvd25sb2FkX2RpcixcbiAgICAgIGZldGNoX2ZpbGVzID0gVFJVRSlcbiAgfVxufSJdfQ== -->

```r
# Download location
download_dir <- file.path(file.path(getwd(), "expression_data"))
if (!dir.exists(download_dir)) {
  dir.create(download_dir) # Create folder if it doesn't exist.
}

# Figure out which files are missing
missing_files <- sfilenames$fname[!unlist(
  lapply(
    sfilenames$fname, FUN=function(x){
      file.exists(file.path(download_dir,data_set_geoid,x))
    }
  )
)]

if(length(missing_files) >0){
  for(i in 1:length(missing_files)) {
    #get the supplementary files
    sfiles = getGEOSuppFiles(data_set_geoid,
      filter_regex = missing_files[i],
      baseDir = download_dir,
      fetch_files = TRUE)
  }
}
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


### Fetching Sample Metadata

The metadata here contains all the information regarding each samples. Data is displayed using knitr [@xieKnitrComprehensiveTool2014].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhIDwtIHJlYWQudGFibGUoXG4gIGZpbGUucGF0aChkb3dubG9hZF9kaXIsZGF0YV9zZXRfZ2VvaWQsIFwiR1NFMTk2NTQ5X0dFTy5tZXRhZGF0YS50c3YuZ3pcIiksXG4gIGhlYWRlcj1UUlVFLFxuICBjaGVjay5uYW1lcz1UUlVFXG4gICkiLCJjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YSA8LSBjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YVssLWMoMSldICMgRGVsZXRlIGR1cGxpY2F0ZSBjb2x1bW4iLCJyb3cubmFtZXMoY29udHJvbGxlcl92c19hcnRfbWV0YWRhdGEpIDwtIGNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhWywxXSIsImRpbShjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YSkiXX0= -->

```r
controller_vs_art_metadata <- read.table(
  file.path(download_dir,data_set_geoid, "GSE196549_GEO.metadata.tsv.gz"),
  header=TRUE,
  check.names=TRUE
  )
controller_vs_art_metadata <- controller_vs_art_metadata[,-c(1)] # Delete duplicate column
row.names(controller_vs_art_metadata) <- controller_vs_art_metadata[,1]
dim(controller_vs_art_metadata)
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIDQ4IDMyXG4ifQ== -->

```
[1] 48 32
```



<!-- rnb-output-end -->

<!-- rnb-source-begin eyJkYXRhIjoia2FibGUoY29udHJvbGxlcl92c19hcnRfbWV0YWRhdGFbMTo4LF0sIGZvcm1hdCA9IFwiaHRtbFwiKSJ9 -->

```r
kable(controller_vs_art_metadata[1:8,], format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> Sample_Name </th>
   <th style="text-align:left;"> Title </th>
   <th style="text-align:right;"> Patient </th>
   <th style="text-align:left;"> Phenotype </th>
   <th style="text-align:left;"> Source </th>
   <th style="text-align:right;"> Replicate </th>
   <th style="text-align:right;"> Batch </th>
   <th style="text-align:left;"> Sample_Well </th>
   <th style="text-align:right;"> Cell_Number </th>
   <th style="text-align:left;"> Response </th>
   <th style="text-align:right;"> Tetramer_Freq </th>
   <th style="text-align:right;"> Age </th>
   <th style="text-align:left;"> Sex </th>
   <th style="text-align:left;"> Race </th>
   <th style="text-align:left;"> Viral_Load </th>
   <th style="text-align:right;"> CD4_Count </th>
   <th style="text-align:right;"> Sample_Years_Since_Diagnosis </th>
   <th style="text-align:right;"> UnalignableReads </th>
   <th style="text-align:right;"> AlignableReads </th>
   <th style="text-align:right;"> TotalReads </th>
   <th style="text-align:right;"> UniqueAlignReads </th>
   <th style="text-align:right;"> MultipleMapReads </th>
   <th style="text-align:right;"> MultiLocReads </th>
   <th style="text-align:right;"> TotalAlign </th>
   <th style="text-align:right;"> AlignRate </th>
   <th style="text-align:right;"> UGPS </th>
   <th style="text-align:right;"> GenesExpCountGRE1 </th>
   <th style="text-align:left;"> FileName.R1 </th>
   <th style="text-align:left;"> FileName.R2 </th>
   <th style="text-align:right;"> pctGC.R1 </th>
   <th style="text-align:right;"> pctGC.R2 </th>
   <th style="text-align:left;"> Sample_Name_Orig </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> R1015.S0031 </td>
   <td style="text-align:left;"> R1015.S0031 </td>
   <td style="text-align:left;"> ART Patient 310817 Source LN 1 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> LN </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> A01 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.2 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 5271449 </td>
   <td style="text-align:right;"> 7606355 </td>
   <td style="text-align:right;"> 12877804 </td>
   <td style="text-align:right;"> 5240615 </td>
   <td style="text-align:right;"> 2365740 </td>
   <td style="text-align:right;"> 6108793 </td>
   <td style="text-align:right;"> 38159750 </td>
   <td style="text-align:right;"> 0.5906562 </td>
   <td style="text-align:right;"> 17194 </td>
   <td style="text-align:right;"> 16937 </td>
   <td style="text-align:left;"> R1015.S0031_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0031_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:left;"> 310817_L1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0032 </td>
   <td style="text-align:left;"> R1015.S0032 </td>
   <td style="text-align:left;"> ART Patient 310817 Source LN 2 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> LN </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> A02 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.2 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 4932287 </td>
   <td style="text-align:right;"> 6181368 </td>
   <td style="text-align:right;"> 11113656 </td>
   <td style="text-align:right;"> 4244114 </td>
   <td style="text-align:right;"> 1937254 </td>
   <td style="text-align:right;"> 4941971 </td>
   <td style="text-align:right;"> 30679697 </td>
   <td style="text-align:right;"> 0.5561957 </td>
   <td style="text-align:right;"> 17081 </td>
   <td style="text-align:right;"> 16819 </td>
   <td style="text-align:left;"> R1015.S0032_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0032_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:left;"> 310817_L2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0033 </td>
   <td style="text-align:left;"> R1015.S0033 </td>
   <td style="text-align:left;"> ART Patient 310817 Source LN 3 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> LN </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> A03 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.2 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 4902886 </td>
   <td style="text-align:right;"> 7393629 </td>
   <td style="text-align:right;"> 12296515 </td>
   <td style="text-align:right;"> 5353554 </td>
   <td style="text-align:right;"> 2040075 </td>
   <td style="text-align:right;"> 5942356 </td>
   <td style="text-align:right;"> 37617645 </td>
   <td style="text-align:right;"> 0.6012784 </td>
   <td style="text-align:right;"> 17845 </td>
   <td style="text-align:right;"> 17526 </td>
   <td style="text-align:left;"> R1015.S0033_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0033_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:left;"> 310817_L3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0034 </td>
   <td style="text-align:left;"> R1015.S0034 </td>
   <td style="text-align:left;"> ART Patient 310817 Source PB 1 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> PB </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> B01 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.5 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 6801967 </td>
   <td style="text-align:right;"> 8361306 </td>
   <td style="text-align:right;"> 15163273 </td>
   <td style="text-align:right;"> 5594620 </td>
   <td style="text-align:right;"> 2766686 </td>
   <td style="text-align:right;"> 6680593 </td>
   <td style="text-align:right;"> 41295547 </td>
   <td style="text-align:right;"> 0.5514183 </td>
   <td style="text-align:right;"> 18143 </td>
   <td style="text-align:right;"> 17873 </td>
   <td style="text-align:left;"> R1015.S0034_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0034_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:left;"> 310817_P1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0035 </td>
   <td style="text-align:left;"> R1015.S0035 </td>
   <td style="text-align:left;"> ART Patient 310817 Source PB 2 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> PB </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> B02 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.5 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 2424310 </td>
   <td style="text-align:right;"> 3506252 </td>
   <td style="text-align:right;"> 5930562 </td>
   <td style="text-align:right;"> 2407407 </td>
   <td style="text-align:right;"> 1098845 </td>
   <td style="text-align:right;"> 2829728 </td>
   <td style="text-align:right;"> 17903216 </td>
   <td style="text-align:right;"> 0.5912175 </td>
   <td style="text-align:right;"> 15252 </td>
   <td style="text-align:right;"> 14987 </td>
   <td style="text-align:left;"> R1015.S0035_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0035_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:left;"> 310817_P2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0036 </td>
   <td style="text-align:left;"> R1015.S0036 </td>
   <td style="text-align:left;"> ART Patient 310817 Source PB 3 </td>
   <td style="text-align:right;"> 310817 </td>
   <td style="text-align:left;"> ART </td>
   <td style="text-align:left;"> PB </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> B03 </td>
   <td style="text-align:right;"> 500 </td>
   <td style="text-align:left;"> B*44 Gag AW11 </td>
   <td style="text-align:right;"> 0.5 </td>
   <td style="text-align:right;"> 55 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 775 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 6874212 </td>
   <td style="text-align:right;"> 9320270 </td>
   <td style="text-align:right;"> 16194482 </td>
   <td style="text-align:right;"> 6429209 </td>
   <td style="text-align:right;"> 2891061 </td>
   <td style="text-align:right;"> 7437627 </td>
   <td style="text-align:right;"> 47114299 </td>
   <td style="text-align:right;"> 0.5755213 </td>
   <td style="text-align:right;"> 18195 </td>
   <td style="text-align:right;"> 17918 </td>
   <td style="text-align:left;"> R1015.S0036_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0036_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:left;"> 310817_P3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0037 </td>
   <td style="text-align:left;"> R1015.S0037 </td>
   <td style="text-align:left;"> EC Patient 642488 Source LN 1 </td>
   <td style="text-align:right;"> 642488 </td>
   <td style="text-align:left;"> EC </td>
   <td style="text-align:left;"> LN </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> C01 </td>
   <td style="text-align:right;"> 2500 </td>
   <td style="text-align:left;"> B*57 Gag KF11 </td>
   <td style="text-align:right;"> 2.4 </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 748 </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:right;"> 5280157 </td>
   <td style="text-align:right;"> 8150687 </td>
   <td style="text-align:right;"> 13430844 </td>
   <td style="text-align:right;"> 5756786 </td>
   <td style="text-align:right;"> 2393901 </td>
   <td style="text-align:right;"> 6535226 </td>
   <td style="text-align:right;"> 40519199 </td>
   <td style="text-align:right;"> 0.6068634 </td>
   <td style="text-align:right;"> 19918 </td>
   <td style="text-align:right;"> 19624 </td>
   <td style="text-align:left;"> R1015.S0037_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0037_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:left;"> 642488_L1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> R1015.S0038 </td>
   <td style="text-align:left;"> R1015.S0038 </td>
   <td style="text-align:left;"> EC Patient 642488 Source LN 2 </td>
   <td style="text-align:right;"> 642488 </td>
   <td style="text-align:left;"> EC </td>
   <td style="text-align:left;"> LN </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 1015 </td>
   <td style="text-align:left;"> C02 </td>
   <td style="text-align:right;"> 2500 </td>
   <td style="text-align:left;"> B*57 Gag KF11 </td>
   <td style="text-align:right;"> 2.4 </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:left;"> Male </td>
   <td style="text-align:left;"> White </td>
   <td style="text-align:left;"> &lt;20 </td>
   <td style="text-align:right;"> 748 </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:right;"> 5755815 </td>
   <td style="text-align:right;"> 8373478 </td>
   <td style="text-align:right;"> 14129293 </td>
   <td style="text-align:right;"> 5987831 </td>
   <td style="text-align:right;"> 2385647 </td>
   <td style="text-align:right;"> 6673607 </td>
   <td style="text-align:right;"> 40761030 </td>
   <td style="text-align:right;"> 0.5926325 </td>
   <td style="text-align:right;"> 19942 </td>
   <td style="text-align:right;"> 19692 </td>
   <td style="text-align:left;"> R1015.S0038_R1_001.fastq.gz </td>
   <td style="text-align:left;"> R1015.S0038_R2_001.fastq.gz </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:left;"> 642488_L2 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Note that there are 3 replicates for each patient. Replicates were consistent across all patients and contributed to the reliability of the data and therefore not manually removed.

### Fetching Sample Data


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImNvbnRyb2xsZXJfdnNfYXJ0X2V4cHIgPC0gcmVhZC50YWJsZShcbiAgZmlsZS5wYXRoKGRvd25sb2FkX2RpcixkYXRhX3NldF9nZW9pZCwgXCJHU0UxOTY1NDlfUjEwMTVSMTIwMC5GSUxULkVDLnRzdi5nelwiKSxcbiAgaGVhZGVyPVRSVUUsXG4gIGNoZWNrLm5hbWVzPVRSVUVcbikiLCJyb3cubmFtZXMoY29udHJvbGxlcl92c19hcnRfZXhwcikgPC0gY29udHJvbGxlcl92c19hcnRfZXhwciRYIiwiY29udHJvbGxlcl92c19hcnRfZXhwciA8LSBjb250cm9sbGVyX3ZzX2FydF9leHByWy1jKDEpXSIsImRpbShjb250cm9sbGVyX3ZzX2FydF9leHByKSJdfQ== -->

```r
controller_vs_art_expr <- read.table(
  file.path(download_dir,data_set_geoid, "GSE196549_R1015R1200.FILT.EC.tsv.gz"),
  header=TRUE,
  check.names=TRUE
)
row.names(controller_vs_art_expr) <- controller_vs_art_expr$X
controller_vs_art_expr <- controller_vs_art_expr[-c(1)]
dim(controller_vs_art_expr)
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIDE4MTYwICAgIDQ4XG4ifQ== -->

```
[1] 18160    48
```



<!-- rnb-output-end -->

<!-- rnb-source-begin eyJkYXRhIjoia2FibGUoY29udHJvbGxlcl92c19hcnRfZXhwclsxOjgsMTo0XSwgZm9ybWF0ID0gXCJodG1sXCIpIn0= -->

```r
kable(controller_vs_art_expr[1:8,1:4], format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> R1015.S0031 </th>
   <th style="text-align:right;"> R1015.S0032 </th>
   <th style="text-align:right;"> R1015.S0033 </th>
   <th style="text-align:right;"> R1015.S0034 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000000003_TSPAN6 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.65 </td>
   <td style="text-align:right;"> 1.05 </td>
   <td style="text-align:right;"> 2.02 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000419_DPM1 </td>
   <td style="text-align:right;"> 312.00 </td>
   <td style="text-align:right;"> 513.01 </td>
   <td style="text-align:right;"> 412.00 </td>
   <td style="text-align:right;"> 337.01 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000457_SCYL3 </td>
   <td style="text-align:right;"> 312.93 </td>
   <td style="text-align:right;"> 235.87 </td>
   <td style="text-align:right;"> 583.01 </td>
   <td style="text-align:right;"> 429.71 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000460_C1orf112 </td>
   <td style="text-align:right;"> 27.95 </td>
   <td style="text-align:right;"> 32.18 </td>
   <td style="text-align:right;"> 95.67 </td>
   <td style="text-align:right;"> 28.06 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000938_FGR </td>
   <td style="text-align:right;"> 840.82 </td>
   <td style="text-align:right;"> 538.40 </td>
   <td style="text-align:right;"> 521.97 </td>
   <td style="text-align:right;"> 780.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000971_CFH </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 58.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 40.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001036_FUCA2 </td>
   <td style="text-align:right;"> 80.00 </td>
   <td style="text-align:right;"> 29.92 </td>
   <td style="text-align:right;"> 149.87 </td>
   <td style="text-align:right;"> 118.06 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001084_GCLC </td>
   <td style="text-align:right;"> 15.00 </td>
   <td style="text-align:right;"> 8.52 </td>
   <td style="text-align:right;"> 70.05 </td>
   <td style="text-align:right;"> 100.01 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Dataset chosen based on the information provided when looking at the materials and methods information above.

### Checking for Identical Expression Values


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImR1cGxpY2F0ZWRfZXhwcmVzc2lvbl9zZWxlY3QgPC0gZHVwbGljYXRlZChjb250cm9sbGVyX3ZzX2FydF9leHByKSIsImR1cGxpY2F0ZWRfZXhwcmVzc2lvbnMgPC0gY29udHJvbGxlcl92c19hcnRfZXhwcltkdXBsaWNhdGVkX2V4cHJlc3Npb25fc2VsZWN0LF0iLCJrYWJsZShkdXBsaWNhdGVkX2V4cHJlc3Npb25zLCBmb3JtYXQgPSBcImh0bWxcIikiXX0= -->

```r
duplicated_expression_select <- duplicated(controller_vs_art_expr)
duplicated_expressions <- controller_vs_art_expr[duplicated_expression_select,]
kable(duplicated_expressions, format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> R1015.S0031 </th>
   <th style="text-align:right;"> R1015.S0032 </th>
   <th style="text-align:right;"> R1015.S0033 </th>
   <th style="text-align:right;"> R1015.S0034 </th>
   <th style="text-align:right;"> R1015.S0035 </th>
   <th style="text-align:right;"> R1015.S0036 </th>
   <th style="text-align:right;"> R1015.S0037 </th>
   <th style="text-align:right;"> R1015.S0038 </th>
   <th style="text-align:right;"> R1015.S0039 </th>
   <th style="text-align:right;"> R1015.S0040 </th>
   <th style="text-align:right;"> R1015.S0041 </th>
   <th style="text-align:right;"> R1015.S0042 </th>
   <th style="text-align:right;"> R1015.S0043 </th>
   <th style="text-align:right;"> R1015.S0044 </th>
   <th style="text-align:right;"> R1015.S0045 </th>
   <th style="text-align:right;"> R1015.S0046 </th>
   <th style="text-align:right;"> R1015.S0047 </th>
   <th style="text-align:right;"> R1015.S0048 </th>
   <th style="text-align:right;"> R1200.S0001 </th>
   <th style="text-align:right;"> R1200.S0002 </th>
   <th style="text-align:right;"> R1200.S0003 </th>
   <th style="text-align:right;"> R1200.S0004 </th>
   <th style="text-align:right;"> R1200.S0005 </th>
   <th style="text-align:right;"> R1200.S0006 </th>
   <th style="text-align:right;"> R1200.S0007 </th>
   <th style="text-align:right;"> R1200.S0008 </th>
   <th style="text-align:right;"> R1200.S0009 </th>
   <th style="text-align:right;"> R1200.S0010 </th>
   <th style="text-align:right;"> R1200.S0011 </th>
   <th style="text-align:right;"> R1200.S0012 </th>
   <th style="text-align:right;"> R1200.S0013 </th>
   <th style="text-align:right;"> R1200.S0014 </th>
   <th style="text-align:right;"> R1200.S0015 </th>
   <th style="text-align:right;"> R1200.S0016 </th>
   <th style="text-align:right;"> R1200.S0017 </th>
   <th style="text-align:right;"> R1200.S0018 </th>
   <th style="text-align:right;"> R1200.S0019 </th>
   <th style="text-align:right;"> R1200.S0020 </th>
   <th style="text-align:right;"> R1200.S0021 </th>
   <th style="text-align:right;"> R1200.S0022 </th>
   <th style="text-align:right;"> R1200.S0023 </th>
   <th style="text-align:right;"> R1200.S0024 </th>
   <th style="text-align:right;"> R1200.S0025 </th>
   <th style="text-align:right;"> R1200.S0026 </th>
   <th style="text-align:right;"> R1200.S0027 </th>
   <th style="text-align:right;"> R1200.S0028 </th>
   <th style="text-align:right;"> R1200.S0029 </th>
   <th style="text-align:right;"> R1200.S0030 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000238122_RP11-483I13.2 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 2.97 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 7.00 </td>
   <td style="text-align:right;"> 9.29 </td>
   <td style="text-align:right;"> 39.50 </td>
   <td style="text-align:right;"> 8.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.96 </td>
   <td style="text-align:right;"> 23.17 </td>
   <td style="text-align:right;"> 43.09 </td>
   <td style="text-align:right;"> 36.48 </td>
   <td style="text-align:right;"> 40.56 </td>
   <td style="text-align:right;"> 47.27 </td>
   <td style="text-align:right;"> 20.88 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.5 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 0.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000274917_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000275215_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000275757_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000276700_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000277739_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000278189_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000278233_RNA5-8S5 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 4.88 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 4.12 </td>
   <td style="text-align:right;"> 8.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 7.25 </td>
   <td style="text-align:right;"> 4.75 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 12.38 </td>
   <td style="text-align:right;"> 7.38 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 7.62 </td>
   <td style="text-align:right;"> 13.12 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 14.62 </td>
   <td style="text-align:right;"> 6.25 </td>
   <td style="text-align:right;"> 5.38 </td>
   <td style="text-align:right;"> 9.38 </td>
   <td style="text-align:right;"> 5.88 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 6.75 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 8.25 </td>
   <td style="text-align:right;"> 2.38 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 4.25 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 3.25 </td>
   <td style="text-align:right;"> 4.62 </td>
   <td style="text-align:right;"> 1.75 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 6.0 </td>
   <td style="text-align:right;"> 1.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 2.12 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 6.38 </td>
   <td style="text-align:right;"> 1.38 </td>
   <td style="text-align:right;"> 3.88 </td>
   <td style="text-align:right;"> 12.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000279245_FAM223A </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 13.00 </td>
   <td style="text-align:right;"> 1.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 3.00 </td>
   <td style="text-align:right;"> 8.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 4.50 </td>
   <td style="text-align:right;"> 1.50 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 3.00 </td>
   <td style="text-align:right;"> 6.50 </td>
   <td style="text-align:right;"> 14.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 3.00 </td>
   <td style="text-align:right;"> 10.50 </td>
   <td style="text-align:right;"> 6.50 </td>
   <td style="text-align:right;"> 7.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 0.50 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 4.00 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.50 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 3.50 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 1.50 </td>
   <td style="text-align:right;"> 4.00 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 8.50 </td>
   <td style="text-align:right;"> 11.0 </td>
   <td style="text-align:right;"> 6.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 13.50 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 10.00 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 6.00 </td>
   <td style="text-align:right;"> 12.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000279654_AL135745.1 </td>
   <td style="text-align:right;"> 0.24 </td>
   <td style="text-align:right;"> 1.45 </td>
   <td style="text-align:right;"> 1.24 </td>
   <td style="text-align:right;"> 1.06 </td>
   <td style="text-align:right;"> 0.48 </td>
   <td style="text-align:right;"> 0.14 </td>
   <td style="text-align:right;"> 2.54 </td>
   <td style="text-align:right;"> 0.81 </td>
   <td style="text-align:right;"> 1.49 </td>
   <td style="text-align:right;"> 5.92 </td>
   <td style="text-align:right;"> 2.15 </td>
   <td style="text-align:right;"> 5.53 </td>
   <td style="text-align:right;"> 1.81 </td>
   <td style="text-align:right;"> 6.06 </td>
   <td style="text-align:right;"> 3.54 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 12.23 </td>
   <td style="text-align:right;"> 3.09 </td>
   <td style="text-align:right;"> 2.55 </td>
   <td style="text-align:right;"> 2.64 </td>
   <td style="text-align:right;"> 1.06 </td>
   <td style="text-align:right;"> 2.62 </td>
   <td style="text-align:right;"> 0.84 </td>
   <td style="text-align:right;"> 1.30 </td>
   <td style="text-align:right;"> 0.22 </td>
   <td style="text-align:right;"> 4.82 </td>
   <td style="text-align:right;"> 7.36 </td>
   <td style="text-align:right;"> 2.08 </td>
   <td style="text-align:right;"> 3.36 </td>
   <td style="text-align:right;"> 2.80 </td>
   <td style="text-align:right;"> 0.52 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.86 </td>
   <td style="text-align:right;"> 0.81 </td>
   <td style="text-align:right;"> 1.49 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 22.64 </td>
   <td style="text-align:right;"> 3.12 </td>
   <td style="text-align:right;"> 2.5 </td>
   <td style="text-align:right;"> 1.68 </td>
   <td style="text-align:right;"> 5.94 </td>
   <td style="text-align:right;"> 9.45 </td>
   <td style="text-align:right;"> 9.39 </td>
   <td style="text-align:right;"> 0.01 </td>
   <td style="text-align:right;"> 16.27 </td>
   <td style="text-align:right;"> 2.09 </td>
   <td style="text-align:right;"> 3.56 </td>
   <td style="text-align:right;"> 24.14 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000280441_CH507-528H12.1 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 13.50 </td>
   <td style="text-align:right;"> 11.50 </td>
   <td style="text-align:right;"> 27.32 </td>
   <td style="text-align:right;"> 7.00 </td>
   <td style="text-align:right;"> 29.52 </td>
   <td style="text-align:right;"> 18.99 </td>
   <td style="text-align:right;"> 13.00 </td>
   <td style="text-align:right;"> 6.00 </td>
   <td style="text-align:right;"> 20.50 </td>
   <td style="text-align:right;"> 13.42 </td>
   <td style="text-align:right;"> 16.79 </td>
   <td style="text-align:right;"> 30.00 </td>
   <td style="text-align:right;"> 26.02 </td>
   <td style="text-align:right;"> 30.50 </td>
   <td style="text-align:right;"> 30.50 </td>
   <td style="text-align:right;"> 11.99 </td>
   <td style="text-align:right;"> 24.48 </td>
   <td style="text-align:right;"> 9.50 </td>
   <td style="text-align:right;"> 7.50 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 12.50 </td>
   <td style="text-align:right;"> 7.00 </td>
   <td style="text-align:right;"> 11.50 </td>
   <td style="text-align:right;"> 7.50 </td>
   <td style="text-align:right;"> 7.37 </td>
   <td style="text-align:right;"> 6.50 </td>
   <td style="text-align:right;"> 7.13 </td>
   <td style="text-align:right;"> 8.50 </td>
   <td style="text-align:right;"> 6.12 </td>
   <td style="text-align:right;"> 6.50 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 4.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 15.84 </td>
   <td style="text-align:right;"> 7.50 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 1.50 </td>
   <td style="text-align:right;"> 4.5 </td>
   <td style="text-align:right;"> 7.50 </td>
   <td style="text-align:right;"> 11.02 </td>
   <td style="text-align:right;"> 4.50 </td>
   <td style="text-align:right;"> 26.50 </td>
   <td style="text-align:right;"> 33.00 </td>
   <td style="text-align:right;"> 14.00 </td>
   <td style="text-align:right;"> 10.00 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 12.50 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000280800_CH507-513H4.6 </td>
   <td style="text-align:right;"> 276768.83 </td>
   <td style="text-align:right;"> 231839.30 </td>
   <td style="text-align:right;"> 154273.36 </td>
   <td style="text-align:right;"> 380449.19 </td>
   <td style="text-align:right;"> 122406.88 </td>
   <td style="text-align:right;"> 305378.11 </td>
   <td style="text-align:right;"> 239747.65 </td>
   <td style="text-align:right;"> 222887.44 </td>
   <td style="text-align:right;"> 162483.88 </td>
   <td style="text-align:right;"> 217713.32 </td>
   <td style="text-align:right;"> 248255.34 </td>
   <td style="text-align:right;"> 169473.71 </td>
   <td style="text-align:right;"> 339851.21 </td>
   <td style="text-align:right;"> 283715.55 </td>
   <td style="text-align:right;"> 459487.99 </td>
   <td style="text-align:right;"> 580859.41 </td>
   <td style="text-align:right;"> 369316.26 </td>
   <td style="text-align:right;"> 472592.50 </td>
   <td style="text-align:right;"> 495618.97 </td>
   <td style="text-align:right;"> 379647.69 </td>
   <td style="text-align:right;"> 287277.73 </td>
   <td style="text-align:right;"> 354812.46 </td>
   <td style="text-align:right;"> 267324.23 </td>
   <td style="text-align:right;"> 306637.98 </td>
   <td style="text-align:right;"> 307843.08 </td>
   <td style="text-align:right;"> 333741.52 </td>
   <td style="text-align:right;"> 304038.44 </td>
   <td style="text-align:right;"> 406570.86 </td>
   <td style="text-align:right;"> 260371.76 </td>
   <td style="text-align:right;"> 273221.75 </td>
   <td style="text-align:right;"> 225732.76 </td>
   <td style="text-align:right;"> 345086.70 </td>
   <td style="text-align:right;"> 222490.80 </td>
   <td style="text-align:right;"> 343296.71 </td>
   <td style="text-align:right;"> 403397.87 </td>
   <td style="text-align:right;"> 232612.18 </td>
   <td style="text-align:right;"> 542895.12 </td>
   <td style="text-align:right;"> 587868.09 </td>
   <td style="text-align:right;"> 464775.0 </td>
   <td style="text-align:right;"> 450415.43 </td>
   <td style="text-align:right;"> 559588.35 </td>
   <td style="text-align:right;"> 472849.26 </td>
   <td style="text-align:right;"> 474857.33 </td>
   <td style="text-align:right;"> 530667.40 </td>
   <td style="text-align:right;"> 445761.93 </td>
   <td style="text-align:right;"> 474882.30 </td>
   <td style="text-align:right;"> 567236.57 </td>
   <td style="text-align:right;"> 777851.11 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000281179_RP11-294C11.4 </td>
   <td style="text-align:right;"> 2.50 </td>
   <td style="text-align:right;"> 3.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 6.43 </td>
   <td style="text-align:right;"> 4.50 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 16.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 9.50 </td>
   <td style="text-align:right;"> 30.50 </td>
   <td style="text-align:right;"> 9.50 </td>
   <td style="text-align:right;"> 2.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 16.00 </td>
   <td style="text-align:right;"> 10.50 </td>
   <td style="text-align:right;"> 12.00 </td>
   <td style="text-align:right;"> 36.50 </td>
   <td style="text-align:right;"> 48.50 </td>
   <td style="text-align:right;"> 36.50 </td>
   <td style="text-align:right;"> 28.50 </td>
   <td style="text-align:right;"> 23.00 </td>
   <td style="text-align:right;"> 21.93 </td>
   <td style="text-align:right;"> 5.50 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 8.50 </td>
   <td style="text-align:right;"> 11.00 </td>
   <td style="text-align:right;"> 5.00 </td>
   <td style="text-align:right;"> 6.50 </td>
   <td style="text-align:right;"> 27.00 </td>
   <td style="text-align:right;"> 18.50 </td>
   <td style="text-align:right;"> 2.0 </td>
   <td style="text-align:right;"> 19.00 </td>
   <td style="text-align:right;"> 15.50 </td>
   <td style="text-align:right;"> 24.50 </td>
   <td style="text-align:right;"> 9.00 </td>
   <td style="text-align:right;"> 16.50 </td>
   <td style="text-align:right;"> 13.50 </td>
   <td style="text-align:right;"> 16.00 </td>
   <td style="text-align:right;"> 3.00 </td>
   <td style="text-align:right;"> 10.50 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000281181_CH507-513H4.3 </td>
   <td style="text-align:right;"> 276768.83 </td>
   <td style="text-align:right;"> 231839.30 </td>
   <td style="text-align:right;"> 154273.36 </td>
   <td style="text-align:right;"> 380449.19 </td>
   <td style="text-align:right;"> 122406.88 </td>
   <td style="text-align:right;"> 305378.11 </td>
   <td style="text-align:right;"> 239747.65 </td>
   <td style="text-align:right;"> 222887.44 </td>
   <td style="text-align:right;"> 162483.88 </td>
   <td style="text-align:right;"> 217713.32 </td>
   <td style="text-align:right;"> 248255.34 </td>
   <td style="text-align:right;"> 169473.71 </td>
   <td style="text-align:right;"> 339851.21 </td>
   <td style="text-align:right;"> 283715.55 </td>
   <td style="text-align:right;"> 459487.99 </td>
   <td style="text-align:right;"> 580859.41 </td>
   <td style="text-align:right;"> 369316.26 </td>
   <td style="text-align:right;"> 472592.50 </td>
   <td style="text-align:right;"> 495618.97 </td>
   <td style="text-align:right;"> 379647.69 </td>
   <td style="text-align:right;"> 287277.73 </td>
   <td style="text-align:right;"> 354812.46 </td>
   <td style="text-align:right;"> 267324.23 </td>
   <td style="text-align:right;"> 306637.98 </td>
   <td style="text-align:right;"> 307843.08 </td>
   <td style="text-align:right;"> 333741.52 </td>
   <td style="text-align:right;"> 304038.44 </td>
   <td style="text-align:right;"> 406570.86 </td>
   <td style="text-align:right;"> 260371.76 </td>
   <td style="text-align:right;"> 273221.75 </td>
   <td style="text-align:right;"> 225732.76 </td>
   <td style="text-align:right;"> 345086.70 </td>
   <td style="text-align:right;"> 222490.80 </td>
   <td style="text-align:right;"> 343296.71 </td>
   <td style="text-align:right;"> 403397.87 </td>
   <td style="text-align:right;"> 232612.18 </td>
   <td style="text-align:right;"> 542895.12 </td>
   <td style="text-align:right;"> 587868.09 </td>
   <td style="text-align:right;"> 464775.0 </td>
   <td style="text-align:right;"> 450415.43 </td>
   <td style="text-align:right;"> 559588.35 </td>
   <td style="text-align:right;"> 472849.26 </td>
   <td style="text-align:right;"> 474857.33 </td>
   <td style="text-align:right;"> 530667.40 </td>
   <td style="text-align:right;"> 445761.93 </td>
   <td style="text-align:right;"> 474882.30 </td>
   <td style="text-align:right;"> 567236.57 </td>
   <td style="text-align:right;"> 777851.11 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


We see several duplicated expressions here. We will remove them.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImNvbnRyb2xsZXJfdnNfYXJ0X2V4cHIgPC0gY29udHJvbGxlcl92c19hcnRfZXhwclshZHVwbGljYXRlZF9leHByZXNzaW9uX3NlbGVjdCxdIiwiZGltKGNvbnRyb2xsZXJfdnNfYXJ0X2V4cHIpIl19 -->

```r
controller_vs_art_expr <- controller_vs_art_expr[!duplicated_expression_select,]
dim(controller_vs_art_expr)
```



<!-- rnb-source-end -->

<!-- rnb-output-begin eyJkYXRhIjoiWzFdIDE4MTQ2ICAgIDQ4XG4ifQ== -->

```
[1] 18146    48
```



<!-- rnb-output-end -->

<!-- rnb-source-begin eyJkYXRhIjoia2FibGUoY29udHJvbGxlcl92c19hcnRfZXhwclsxOjgsMTo0XSwgZm9ybWF0ID0gXCJodG1sXCIpIn0= -->

```r
kable(controller_vs_art_expr[1:8,1:4], format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> R1015.S0031 </th>
   <th style="text-align:right;"> R1015.S0032 </th>
   <th style="text-align:right;"> R1015.S0033 </th>
   <th style="text-align:right;"> R1015.S0034 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000000003_TSPAN6 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.65 </td>
   <td style="text-align:right;"> 1.05 </td>
   <td style="text-align:right;"> 2.02 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000419_DPM1 </td>
   <td style="text-align:right;"> 312.00 </td>
   <td style="text-align:right;"> 513.01 </td>
   <td style="text-align:right;"> 412.00 </td>
   <td style="text-align:right;"> 337.01 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000457_SCYL3 </td>
   <td style="text-align:right;"> 312.93 </td>
   <td style="text-align:right;"> 235.87 </td>
   <td style="text-align:right;"> 583.01 </td>
   <td style="text-align:right;"> 429.71 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000460_C1orf112 </td>
   <td style="text-align:right;"> 27.95 </td>
   <td style="text-align:right;"> 32.18 </td>
   <td style="text-align:right;"> 95.67 </td>
   <td style="text-align:right;"> 28.06 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000938_FGR </td>
   <td style="text-align:right;"> 840.82 </td>
   <td style="text-align:right;"> 538.40 </td>
   <td style="text-align:right;"> 521.97 </td>
   <td style="text-align:right;"> 780.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000971_CFH </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 58.00 </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 40.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001036_FUCA2 </td>
   <td style="text-align:right;"> 80.00 </td>
   <td style="text-align:right;"> 29.92 </td>
   <td style="text-align:right;"> 149.87 </td>
   <td style="text-align:right;"> 118.06 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001084_GCLC </td>
   <td style="text-align:right;"> 15.00 </td>
   <td style="text-align:right;"> 8.52 </td>
   <td style="text-align:right;"> 70.05 </td>
   <td style="text-align:right;"> 100.01 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


## Overview of Data

The following shows the different number of samples for the different sample sources.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhIDwtIGRhdGEudGFibGU6OmRhdGEudGFibGUoY29udHJvbGxlcl92c19hcnRfbWV0YWRhdGEpIiwiY29udHJvbGxlcl92c19hcnRfbWV0YWRhdGFbLCAuKGNvdW50ID0gLk4pLCBieSA9IGNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhJFNvdXJjZV0iXX0= -->

```r
controller_vs_art_metadata <- data.table::data.table(controller_vs_art_metadata)
controller_vs_art_metadata[, .(count = .N), by = controller_vs_art_metadata$Source]
```



<!-- rnb-source-end -->
<div data-pagedtable="false">
  <script data-pagedtable-source type="application/json">
{"columns":[{"label":["controller_vs_art_metadata"],"name":[1],"type":["chr"],"align":["left"]},{"label":["count"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"LN","2":"24"},{"1":"PB","2":"24"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


The following shows the different number of samples for the different phenotypes.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjoiY29udHJvbGxlcl92c19hcnRfbWV0YWRhdGFbLCAuKGNvdW50ID0gLk4pLCBieSA9IGNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhJFBoZW5vdHlwZV0ifQ== -->

```r
controller_vs_art_metadata[, .(count = .N), by = controller_vs_art_metadata$Phenotype]
```



<!-- rnb-source-end -->
<div data-pagedtable="false">
  <script data-pagedtable-source type="application/json">
{"columns":[{"label":["controller_vs_art_metadata"],"name":[1],"type":["chr"],"align":["left"]},{"label":["count"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"ART","2":"6"},{"1":"EC","2":"18"},{"1":"VC","2":"24"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


## Visualizing Expression Data

### Expression Level Distribution Across Samples

Visualizing data via comparing centers with use of Boxplots.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImxvZ19jb250cm9sbGVyX3ZzX2FydF9leHByIDwtIGxvZzIoY29udHJvbGxlcl92c19hcnRfZXhwcikgIyBMb2cyIGFsbCBleHByZXNzaW9uIGNvdW50cyIsIiIsImNlbnRlcl9kZXZpYXRpb25fcGxvdCA8LSBmdW5jdGlvbihncmFwaGVkX2RhdGEsIHRpdGxlKSB7XG4gIGJveHBsb3QoZ3JhcGhlZF9kYXRhLCB4bGFiID0gXCJTYW1wbGVzXCIsIHlsYWIgPSBcIkxvZzIgRXhwci5cIixcbiAgbGFzID0gMiwgY2V4ID0gMC41LCBjZXgubGFiID0gMC41LFxuICBjZXguYXhpcyA9IDAuNSwgbWFpbiA9IHRpdGxlKVxuXG4gIGFibGluZShcbiAgICBoID0gbWVkaWFuKGFwcGx5KGdyYXBoZWRfZGF0YSwgMiwgbWVkaWFuKSksXG4gICAgY29sID0gXCJkYXJrcmVkXCIsXG4gICAgbHdkID0gMSxcbiAgICBsdHkgPSBcImRhc2hlZFwiXG4gIClcbn0iLCIiLCJjZW50ZXJfZGV2aWF0aW9uX3Bsb3QobG9nX2NvbnRyb2xsZXJfdnNfYXJ0X2V4cHIsIFwiQ0Q4KyBUIENlbGwgR2VuZSBFeHByZXNzaW9uIGluIEhJViBQYXRpZW50c1wiKSJdfQ== -->

```r
log_controller_vs_art_expr <- log2(controller_vs_art_expr) # Log2 all expression counts

center_deviation_plot <- function(graphed_data, title) {
  boxplot(graphed_data, xlab = "Samples", ylab = "Log2 Expr.",
  las = 2, cex = 0.5, cex.lab = 0.5,
  cex.axis = 0.5, main = title)

  abline(
    h = median(apply(graphed_data, 2, median)),
    col = "darkred",
    lwd = 1,
    lty = "dashed"
  )
}

center_deviation_plot(log_controller_vs_art_expr, "CD8+ T Cell Gene Expression in HIV Patients")
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-22-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


These series of boxplots shows the distribution of expressions across all genes and the skew of the data for each sample taken. The darkgreen line shows the median across all the data points. To elaborate, horizontally, each sample is presented and labelled. Vertically is the $log_2$ of the expression levels for the variety of genes.

From this, we can see that there are a few samples that have wildly different expression distributions compared to others. This could indicate minor errors or noise in the data whether this be technically introduced or biological cause.

### Density

Continuing visualization with density plot.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbIiMgVE9ETyBjaGVjayBsYWJlbHMiLCIiLCJkZW5zaXR5X3Bsb3QgPC0gZnVuY3Rpb24oZ3JhcGhlZF9kYXRhLCB0aXRsZSkge1xuICBjb3VudHNfZGVuc2l0eSA8LSBhcHBseShsb2cyKGdyYXBoZWRfZGF0YSksIDIsIGRlbnNpdHkpXG4gICNjYWxjdWxhdGUgdGhlIGxpbWl0cyBhY3Jvc3MgYWxsIHRoZSBzYW1wbGVzXG4gIHhsaW0gPC0gMDsgeWxpbSA8LSAwXG4gIGZvciAoaSBpbiAxOmxlbmd0aChjb3VudHNfZGVuc2l0eSkpIHtcbiAgICB4bGltIDwtIHJhbmdlKGMoeGxpbSwgY291bnRzX2RlbnNpdHlbW2ldXSR4KSk7XG4gICAgeWxpbSA8LSByYW5nZShjKHlsaW0sIGNvdW50c19kZW5zaXR5W1tpXV0keSkpXG4gIH1cblxuICBjb2xzIDwtIHJhaW5ib3cobGVuZ3RoKGNvdW50c19kZW5zaXR5KSlcbiAgbHR5cyA8LSByZXAoMSwgbGVuZ3RoKGNvdW50c19kZW5zaXR5KSlcblxuICAjcGxvdCB0aGUgZmlyc3QgZGVuc2l0eSBwbG90IHRvIGluaXRpYWxpemUgdGhlIHBsb3RcbiAgcGxvdChjb3VudHNfZGVuc2l0eVtbMV1dLCB4bGltPXhsaW0sIHlsaW09eWxpbSwgdHlwZT1cIm5cIixcbiAgICB5bGFiPVwiU21vb3RoaW5nIGRlbnNpdHkgb2YgbG9nMi1DUE1cIixcbiAgICBtYWluPXRpdGxlLCBjZXgubGFiID0gMC44NSlcblxuICAjcGxvdCBlYWNoIGxpbmVcbiAgZm9yIChpIGluIDE6bGVuZ3RoKGNvdW50c19kZW5zaXR5KSlcbiAgICBsaW5lcyhjb3VudHNfZGVuc2l0eVtbaV1dLCBjb2w9Y29sc1tpXSwgbHR5PWx0eXNbaV0pXG5cbiAgI2NyZWF0ZSBsZWdlbmRcbiAgbGVnZW5kKFwidG9wcmlnaHRcIiwgY29sbmFtZXMoZ3JhcGhlZF9kYXRhKSxcbiAgICBjb2w9Y29scywgbHR5PWx0eXMsIGNleD0wLjc1LFxuICAgIGJvcmRlciA9XCJibHVlXCIsIHRleHQuY29sID0gXCJncmVlbjRcIixcbiAgICBtZXJnZSA9IFRSVUUsIGJnID0gXCJncmF5OTBcIilcbn0iLCIiLCJkZW5zaXR5X3Bsb3QoY29udHJvbGxlcl92c19hcnRfZXhwciwgXCJFeHByZXNzaW9uIERlbnNpdHkgQ3VydmVcIikiXX0= -->

```r
# TODO check labels

density_plot <- function(graphed_data, title) {
  counts_density <- apply(log2(graphed_data), 2, density)
  #calculate the limits across all the samples
  xlim <- 0; ylim <- 0
  for (i in 1:length(counts_density)) {
    xlim <- range(c(xlim, counts_density[[i]]$x));
    ylim <- range(c(ylim, counts_density[[i]]$y))
  }

  cols <- rainbow(length(counts_density))
  ltys <- rep(1, length(counts_density))

  #plot the first density plot to initialize the plot
  plot(counts_density[[1]], xlim=xlim, ylim=ylim, type="n",
    ylab="Smoothing density of log2-CPM",
    main=title, cex.lab = 0.85)

  #plot each line
  for (i in 1:length(counts_density))
    lines(counts_density[[i]], col=cols[i], lty=ltys[i])

  #create legend
  legend("topright", colnames(graphed_data),
    col=cols, lty=ltys, cex=0.75,
    border ="blue", text.col = "green4",
    merge = TRUE, bg = "gray90")
}

density_plot(controller_vs_art_expr, "Expression Density Curve")
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-23-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


This density plot shows the density of different $log_2$ expression levels for different genes where the smoothed density is displayed on the vertical axis and the value corresponding to the density is shown on the horizontal axis. The different colours represent the different samples.

From this visualization, we can see there seems to be certain levels of expression that have a relatively larger variance between different samples. This ultimately agrees with the previous boxplot result in that there may be noise in these regions.

## Filtering the dataset

We will begin attempting to clean up the variations seen in the data across different samples by requiring a minimum of $30$ samples where the CPM of the expression value is greater than $1$.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbIm1pbl9udW1fc2FtcGxlcyA8LSAzMCIsImRhdGFfbWF0cml4IDwtIGFzLm1hdHJpeChjb250cm9sbGVyX3ZzX2FydF9leHByKSIsIiIsIiMgZ2V0IHJpZCBvZiBsb3cgY291bnRzIiwiZmlsdGVyZWRfY29udHJvbGxlcl92c19hcnRfZXhwciA9IGRhdGFfbWF0cml4W1xuICByb3dTdW1zKGNwbShkYXRhX21hdHJpeCkgPjEpID4gbWluX251bV9zYW1wbGVzXG4sXSIsIiIsInBhcihtZnJvdz1jKDEsIDIpKSIsImRlbnNpdHlfcGxvdChjb250cm9sbGVyX3ZzX2FydF9leHByLCBcIk9yaWdpbmFsXCIpIiwiZGVuc2l0eV9wbG90KGZpbHRlcmVkX2NvbnRyb2xsZXJfdnNfYXJ0X2V4cHIsIFwiQWZ0ZXIgRmlsdGVyaW5nXCIpIl19 -->

```r
min_num_samples <- 30
data_matrix <- as.matrix(controller_vs_art_expr)

# get rid of low counts
filtered_controller_vs_art_expr = data_matrix[
  rowSums(cpm(data_matrix) >1) > min_num_samples
,]

par(mfrow=c(1, 2))
density_plot(controller_vs_art_expr, "Original")
density_plot(filtered_controller_vs_art_expr, "After Filtering")
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-24-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-source-begin eyJkYXRhIjoicGFyKG1mcm93PWMoMSwgMSkpIn0= -->

```r
par(mfrow=c(1, 1))
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


By doing so, we see we've reduced the variation at the points where there is an increase in differences between different samples in terms of expression level density. However, there is still quite a bit of variation in general in terms of density distribution across samples. A total of 6416 was removed due to filtering. The exact statistical method of removing outliers was not described in detail by the original paper [@collinsCytolyticCD8Cells2023].

## Applying TMM

We will now apply trimmed mean of m-values or TMM to normalize the data further. We choose to apply TMM as we expect that most genes are not differentially. expressed. Furthermore, as the process to procure this data contains many technically based areas of potentially introducing variation, TMM may help by trimming off extreme values likely caused by technical variation.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbInBoZW5vdHlwZV9kZ2UgPC0gREdFTGlzdChjb3VudHM9ZmlsdGVyZWRfY29udHJvbGxlcl92c19hcnRfZXhwciwgZ3JvdXA9Y29udHJvbGxlcl92c19hcnRfbWV0YWRhdGEkUGhlbm90eXBlKSIsInBoZW5vdHlwZV9kZ2UgPC0gY2FsY05vcm1GYWN0b3JzKHBoZW5vdHlwZV9kZ2UpIiwibm9ybWVkX2NvbnRyb2xsZXJfdnNfYXJ0IDwtIGNwbShwaGVub3R5cGVfZGdlKSJdfQ== -->

```r
phenotype_dge <- DGEList(counts=filtered_controller_vs_art_expr, group=controller_vs_art_metadata$Phenotype)
phenotype_dge <- calcNormFactors(phenotype_dge)
normed_controller_vs_art <- cpm(phenotype_dge)
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->



<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbInBhcihtZnJvdz1jKDIsIDIpKSIsImRlbnNpdHlfcGxvdChmaWx0ZXJlZF9jb250cm9sbGVyX3ZzX2FydF9leHByLCBcIk9yaWdpbmFsXCIpIiwiZGVuc2l0eV9wbG90KG5vcm1lZF9jb250cm9sbGVyX3ZzX2FydCwgXCJUTU1cIikiLCJjZW50ZXJfZGV2aWF0aW9uX3Bsb3QobG9nMihmaWx0ZXJlZF9jb250cm9sbGVyX3ZzX2FydF9leHByKSwgXCJPcmlnaW5hbFwiKSJdfQ== -->

```r
par(mfrow=c(2, 2))
density_plot(filtered_controller_vs_art_expr, "Original")
density_plot(normed_controller_vs_art, "TMM")
center_deviation_plot(log2(filtered_controller_vs_art_expr), "Original")
```



<!-- rnb-source-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 1 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 2 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 3 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 4 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA1IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 5 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA2IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 6 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA3IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 7 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA4IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 8 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA5IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 9 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 10 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 11 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 12 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 13 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 14 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 15 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 16 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 17 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 18 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 19 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 20 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 21 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 22 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 23 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 24 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 25 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 26 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 27 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 28 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 29 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 30 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 31 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 32 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 33 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 34 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 35 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 36 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 37 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 38 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 39 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 40 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 41 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 42 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 43 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 44 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 45 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 46 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 47 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0OCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 48 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-source-begin eyJkYXRhIjoiY2VudGVyX2RldmlhdGlvbl9wbG90KGxvZzIobm9ybWVkX2NvbnRyb2xsZXJfdnNfYXJ0KSwgXCJUTU1cIikifQ== -->

```r
center_deviation_plot(log2(normed_controller_vs_art), "TMM")
```



<!-- rnb-source-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 1 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 2 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzIGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 3 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 4 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA1IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 5 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA2IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 6 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA3IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 7 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA4IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 8 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA5IGlzIG5vdCBkcmF3blxuIn0= -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 9 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 10 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 11 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 12 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 13 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 14 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 15 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 16 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 17 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 18 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAxOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 19 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 20 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 21 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 22 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 23 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 24 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 25 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 26 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 27 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 28 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAyOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 29 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 30 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 31 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 32 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzMyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 33 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 34 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 35 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 36 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzNyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 37 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzOCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 38 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCAzOSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 39 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 40 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 41 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 42 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0MyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 43 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 44 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NSBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 45 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NiBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 46 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0NyBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 47 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-warning-begin eyJkYXRhIjoiV2FybmluZyBpbiBicGx0KGF0W2ldLCB3aWQgPSB3aWR0aFtpXSwgc3RhdHMgPSB6JHN0YXRzWywgaV0sIG91dCA9XG56JG91dFt6JGdyb3VwID09IDogT3V0bGllciAoLUluZikgaW4gYm94cGxvdCA0OCBpcyBub3QgZHJhd25cbiJ9 -->

```
Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out =
z$out[z$group == : Outlier (-Inf) in boxplot 48 is not drawn
```



<!-- rnb-warning-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-26-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-source-begin eyJkYXRhIjoicGFyKG1mcm93PWMoMSwgMSkpIn0= -->

```r
par(mfrow=c(1, 1))
```



<!-- rnb-source-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Completing the TMM show a vast improvement in terms of variation in distribution of expression levels and the centers of the expression levels.

## Sample Difference visualization

We can visualize the difference between samples via plotting them on a multidimensional scaling (MDS) plot.

To start, we will color the points first based on the source of the sample. The MDS plot will be generated using the limma package [@ritchieLimmaPowersDifferential2015].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbInBsb3RNRFMocGhlbm90eXBlX2RnZSwgbGFiZWxzPU5VTEwsIHBjaCA9IDEsIGNvbCA9IGMoXCJyZWRcIixcImJsdWVcIilbZmFjdG9yKGNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhJFNvdXJjZSldKSIsImxlZ2VuZChcInRvcHJpZ2h0XCIsXG4gIGxlZ2VuZD1sZXZlbHMoZmFjdG9yKGNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhJFNvdXJjZSkpLFxuICBwY2g9YygxKSwgY29sPSBjKFwicmVkXCIsXCJibHVlXCIpLHRpdGxlPVwiQ2xhc3NcIixcbiAgYnR5ID0gJ24nLCBjZXggPSAwLjc1KSJdfQ== -->

```r
plotMDS(phenotype_dge, labels=NULL, pch = 1, col = c("red","blue")[factor(controller_vs_art_metadata$Source)])
legend("topright",
  legend=levels(factor(controller_vs_art_metadata$Source)),
  pch=c(1), col= c("red","blue"),title="Class",
  bty = 'n', cex = 0.75)
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-27-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Looking at the data categorized by source, we see no substantial differences between the sources in that where there are lymph node (LN) samples, there are also peripheral blood (PB) samples in a similar area.

To continue, let's color based on the phenotype of the sample.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbInBsb3RNRFMocGhlbm90eXBlX2RnZSwgbGFiZWxzPU5VTEwscGNoPTEsIGNvbCA9IGMoXCJyZWRcIixcIm9yYW5nZVwiLFwicHVycGxlXCIpW2ZhY3Rvcihjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YSRQaGVub3R5cGUpXSkiLCJsZWdlbmQoXCJ0b3ByaWdodFwiLFxuICBsZWdlbmQ9bGV2ZWxzKGZhY3Rvcihjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YSRQaGVub3R5cGUpKSxcbiAgcGNoPWMoMSksIGNvbD1cbmMoXCJyZWRcIixcIm9yYW5nZVwiLFwicHVycGxlXCIpLHRpdGxlPVwiQ2xhc3NcIixcbiAgYnR5ID0gJ24nLCBjZXggPSAwLjc1KSJdfQ== -->

```r
plotMDS(phenotype_dge, labels=NULL,pch=1, col = c("red","orange","purple")[factor(controller_vs_art_metadata$Phenotype)])
legend("topright",
  legend=levels(factor(controller_vs_art_metadata$Phenotype)),
  pch=c(1), col=
c("red","orange","purple"),title="Class",
  bty = 'n', cex = 0.75)
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-28-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


In stark contrast, to the previous color categories, it seems that there are forms of clusters between ART and EC. Furthermore, a large portion of the viremic controllers are also clustered by themselves apart from the other two clusters. Lastly, there exists a third cluster comprised of samples from EC patients, and controls.

Based on this, there seems to be a group of EC that have very similar expression levels compared to patients on ART, however, there may be another group of patients that are categorically elite controllers that have similar gene expressions as a small portion of the viremic control results. This could indicate that there are biological variables that are not captured by this experiment which differentiate the control and the EC.

Interpreting results was based on separate review material [@fullerMolecularClassificationHuman2002].

### Dispersion Analysis

Shows the variability of gene expression relative to the mean for a given counts per million (CPM). Biological Coefficient of Variation (BCV) was plotted with edgeR [@chenEdgeRPowerfulDifferential2024].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbIm1vZGVsX2Rlc2lnbiA8LSBtb2RlbC5tYXRyaXgofmNvbnRyb2xsZXJfdnNfYXJ0X21ldGFkYXRhJFBoZW5vdHlwZStjb250cm9sbGVyX3ZzX2FydF9tZXRhZGF0YSRTb3VyY2UpIiwicGhlbm90eXBlX2RnZSA8LSBlc3RpbWF0ZURpc3AocGhlbm90eXBlX2RnZSwgbW9kZWxfZGVzaWduKSIsInBsb3RCQ1YocGhlbm90eXBlX2RnZSwgY29sLnRhZ3dpc2UgPSBcImJsYWNrXCIsY29sLmNvbW1vbiA9IFwicmVkXCIsKSJdfQ== -->

```r
model_design <- model.matrix(~controller_vs_art_metadata$Phenotype+controller_vs_art_metadata$Source)
phenotype_dge <- estimateDisp(phenotype_dge, model_design)
plotBCV(phenotype_dge, col.tagwise = "black",col.common = "red",)
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-29-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


The produced figure indicates that for more highly expressed genes, the biological coefficient of variation decreases. This trend only seems to occur after approximately 3 $\logCPM$. This may further be interpreted to indicating there is a decrease in the number of differential genes as the expression counts increase. The following plot was generated with use of edgeR [@chenEdgeRPowerfulDifferential2024].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjoicGxvdE1lYW5WYXIocGhlbm90eXBlX2RnZSwgc2hvdy5yYXcudmFycyA9IFRSVUUsXG4gIHNob3cudGFnd2lzZS52YXJzPVRSVUUsIE5CbGluZT1UUlVFLFxuICBzaG93LmF2ZS5yYXcudmFycyA9IFRSVUUsc2hvdy5iaW5uZWQuY29tbW9uLmRpc3AudmFyc1xuPSBUUlVFKSJ9 -->

```r
plotMeanVar(phenotype_dge, show.raw.vars = TRUE,
  show.tagwise.vars=TRUE, NBline=TRUE,
  show.ave.raw.vars = TRUE,show.binned.common.disp.vars
= TRUE)
```



<!-- rnb-source-end -->

<!-- rnb-plot-begin -->

<img src="/home/coder/.vscode-R/tmp/7b2bcd8d66e920d5d4208317b396629e93ccbffe8078c2c7ba9cfbe056375de7_files/figure-html/unnamed-chunk-30-1.png" width="672" />

<!-- rnb-plot-end -->

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


This figure confirms the assumption that the data follows the negatively binomial distribution much closely. The blue line shows the negative binomial distribution.

## (Re)Mapping HUGO Symbols

We will remap all given Ensembl Stable IDs to HUGO maps to ensure IDs are correct. Parsing given naming scheme will require use of stringr [@wickhamStringrSimpleConsistent2023]. Fetching based on Stable Ensembl gene IDs to retrieve HUGO symbols will be done through the biomaRt package [@durinckBioMartBioconductorPowerful2005].


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbImVuc2VtYmwgPSB1c2VEYXRhc2V0KFwiaHNhcGllbnNfZ2VuZV9lbnNlbWJsXCIsbWFydD11c2VNYXJ0KFwiZW5zZW1ibFwiKSkiLCIiLCIiLCJpZHMyY29udmVydCA8LSBsYXBwbHkocm93bmFtZXMobm9ybWVkX2NvbnRyb2xsZXJfdnNfYXJ0KSwgRlVOID0gZnVuY3Rpb24oeCkge1xuICByZXR1cm4odW5saXN0KHN0cl9zcGxpdCh4LCBcIl9cIikpW1sxXV0pXG59KSIsIiIsInJvd25hbWVzKG5vcm1lZF9jb250cm9sbGVyX3ZzX2FydCkgPC0gaWRzMmNvbnZlcnQiLCIiLCJjb252ZXJzaW9uX2NhY2hlIDwtIFwiaWRfY29udmVyc2lvbi5yZHNcIiIsIiIsImNvbnZlcnNpb25fY2FjaGVfZmlsZSA8LSBcImlkX2NvbnZlcnNpb24ucmRzXCIiLCJpZiAoZmlsZS5leGlzdHMoY29udmVyc2lvbl9jYWNoZV9maWxlKSkge1xuICBpZF9jb252ZXJzaW9uIDwtIHJlYWRSRFMoY29udmVyc2lvbl9jYWNoZV9maWxlKVxufSBlbHNlIHtcbiAgaWRfY29udmVyc2lvbiA8LSBnZXRCTShhdHRyaWJ1dGVzID1cbiAgICBjKFwiZW5zZW1ibF9nZW5lX2lkXCIsXCJoZ25jX3N5bWJvbFwiKSxcbiAgICBmaWx0ZXJzID0gYyhcImVuc2VtYmxfZ2VuZV9pZFwiKSxcbiAgICB2YWx1ZXMgPSB1bmxpc3QoaWRzMmNvbnZlcnQpLFxuICAgIG1hcnQgPSBlbnNlbWJsXG4gIClcbiAgc2F2ZVJEUyhpZF9jb252ZXJzaW9uLCBjb252ZXJzaW9uX2NhY2hlX2ZpbGUpXG59IiwiIiwicmVhbm5vdF9ub3JtX2NvbnRyb2xsZXJfdnNfYXJ0IDwtIG1lcmdlKFxuICB4ID0gaWRfY29udmVyc2lvbixcbiAgeSA9IG5vcm1lZF9jb250cm9sbGVyX3ZzX2FydCxcbiAgYnkueCA9IDEsXG4gIGJ5LnkgPSAwLFxuICBhbGwueSA9IFRSVUVcbikiLCIiLCJrYWJsZShyZWFubm90X25vcm1fY29udHJvbGxlcl92c19hcnRbMTo4LDE6OF0sIGZvcm1hdCA9IFwiaHRtbFwiKSJdfQ== -->

```r
ensembl = useDataset("hsapiens_gene_ensembl",mart=useMart("ensembl"))


ids2convert <- lapply(rownames(normed_controller_vs_art), FUN = function(x) {
  return(unlist(str_split(x, "_"))[[1]])
})

rownames(normed_controller_vs_art) <- ids2convert

conversion_cache <- "id_conversion.rds"

conversion_cache_file <- "id_conversion.rds"
if (file.exists(conversion_cache_file)) {
  id_conversion <- readRDS(conversion_cache_file)
} else {
  id_conversion <- getBM(attributes =
    c("ensembl_gene_id","hgnc_symbol"),
    filters = c("ensembl_gene_id"),
    values = unlist(ids2convert),
    mart = ensembl
  )
  saveRDS(id_conversion, conversion_cache_file)
}

reannot_norm_controller_vs_art <- merge(
  x = id_conversion,
  y = normed_controller_vs_art,
  by.x = 1,
  by.y = 0,
  all.y = TRUE
)

kable(reannot_norm_controller_vs_art[1:8,1:8], format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;"> ensembl_gene_id </th>
   <th style="text-align:left;"> hgnc_symbol </th>
   <th style="text-align:right;"> R1015.S0031 </th>
   <th style="text-align:right;"> R1015.S0032 </th>
   <th style="text-align:right;"> R1015.S0033 </th>
   <th style="text-align:right;"> R1015.S0034 </th>
   <th style="text-align:right;"> R1015.S0035 </th>
   <th style="text-align:right;"> R1015.S0036 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000000419 </td>
   <td style="text-align:left;"> DPM1 </td>
   <td style="text-align:right;"> 46.449060 </td>
   <td style="text-align:right;"> 92.701302 </td>
   <td style="text-align:right;"> 57.07520 </td>
   <td style="text-align:right;"> 45.465551 </td>
   <td style="text-align:right;"> 71.933686 </td>
   <td style="text-align:right;"> 61.391601 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000457 </td>
   <td style="text-align:left;"> SCYL3 </td>
   <td style="text-align:right;"> 46.587514 </td>
   <td style="text-align:right;"> 42.621890 </td>
   <td style="text-align:right;"> 80.76557 </td>
   <td style="text-align:right;"> 57.971580 </td>
   <td style="text-align:right;"> 44.635428 </td>
   <td style="text-align:right;"> 22.355787 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000460 </td>
   <td style="text-align:left;"> FIRRM </td>
   <td style="text-align:right;"> 4.161062 </td>
   <td style="text-align:right;"> 5.814951 </td>
   <td style="text-align:right;"> 13.25336 </td>
   <td style="text-align:right;"> 3.785536 </td>
   <td style="text-align:right;"> 0.000000 </td>
   <td style="text-align:right;"> 9.928136 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000938 </td>
   <td style="text-align:left;"> FGR </td>
   <td style="text-align:right;"> 125.177238 </td>
   <td style="text-align:right;"> 97.289294 </td>
   <td style="text-align:right;"> 72.30957 </td>
   <td style="text-align:right;"> 105.228717 </td>
   <td style="text-align:right;"> 150.925549 </td>
   <td style="text-align:right;"> 101.517767 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001036 </td>
   <td style="text-align:left;"> FUCA2 </td>
   <td style="text-align:right;"> 11.910015 </td>
   <td style="text-align:right;"> 5.406567 </td>
   <td style="text-align:right;"> 20.76180 </td>
   <td style="text-align:right;"> 15.927311 </td>
   <td style="text-align:right;"> 21.775024 </td>
   <td style="text-align:right;"> 19.452130 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001084 </td>
   <td style="text-align:left;"> GCLC </td>
   <td style="text-align:right;"> 2.233128 </td>
   <td style="text-align:right;"> 1.539571 </td>
   <td style="text-align:right;"> 9.70417 </td>
   <td style="text-align:right;"> 13.492210 </td>
   <td style="text-align:right;"> 5.137891 </td>
   <td style="text-align:right;"> 11.334928 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001167 </td>
   <td style="text-align:left;"> NFYA </td>
   <td style="text-align:right;"> 64.909583 </td>
   <td style="text-align:right;"> 42.826082 </td>
   <td style="text-align:right;"> 56.93944 </td>
   <td style="text-align:right;"> 50.860547 </td>
   <td style="text-align:right;"> 47.525492 </td>
   <td style="text-align:right;"> 25.836620 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000001460 </td>
   <td style="text-align:left;"> STPG1 </td>
   <td style="text-align:right;"> 26.772226 </td>
   <td style="text-align:right;"> 19.419912 </td>
   <td style="text-align:right;"> 13.41406 </td>
   <td style="text-align:right;"> 15.645352 </td>
   <td style="text-align:right;"> 21.495652 </td>
   <td style="text-align:right;"> 26.015580 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


There were a few symbols that were not successfully matched. Let's take a look.


<!-- rnb-text-end -->


<!-- rnb-chunk-begin -->


<!-- rnb-source-begin eyJkYXRhIjpbIm1pc3NpbmdfaHVnb19ub3JtX2NvbnRyb2xsZXJfdnNfYXJ0IDwtIHJlYW5ub3Rfbm9ybV9jb250cm9sbGVyX3ZzX2FydFtcbiAgaXMubmEocmVhbm5vdF9ub3JtX2NvbnRyb2xsZXJfdnNfYXJ0JGhnbmNfc3ltYm9sKSxcbl0iLCJrYWJsZShtaXNzaW5nX2h1Z29fbm9ybV9jb250cm9sbGVyX3ZzX2FydFssMTo4XSwgZm9ybWF0ID0gXCJodG1sXCIpIl19 -->

```r
missing_hugo_norm_controller_vs_art <- reannot_norm_controller_vs_art[
  is.na(reannot_norm_controller_vs_art$hgnc_symbol),
]
kable(missing_hugo_norm_controller_vs_art[,1:8], format = "html")
```



<!-- rnb-source-end -->
<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> ensembl_gene_id </th>
   <th style="text-align:left;"> hgnc_symbol </th>
   <th style="text-align:right;"> R1015.S0031 </th>
   <th style="text-align:right;"> R1015.S0032 </th>
   <th style="text-align:right;"> R1015.S0033 </th>
   <th style="text-align:right;"> R1015.S0034 </th>
   <th style="text-align:right;"> R1015.S0035 </th>
   <th style="text-align:right;"> R1015.S0036 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 2566 </td>
   <td style="text-align:left;"> ENSG00000112096 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 142.5881918 </td>
   <td style="text-align:right;"> 189.3725993 </td>
   <td style="text-align:right;"> 154.6544040 </td>
   <td style="text-align:right;"> 192.7196503 </td>
   <td style="text-align:right;"> 140.1680896 </td>
   <td style="text-align:right;"> 150.1475001 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2977 </td>
   <td style="text-align:left;"> ENSG00000116957 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 20.8425268 </td>
   <td style="text-align:right;"> 12.1069516 </td>
   <td style="text-align:right;"> 15.2385249 </td>
   <td style="text-align:right;"> 20.5061090 </td>
   <td style="text-align:right;"> 10.5969002 </td>
   <td style="text-align:right;"> 9.7183615 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 3913 </td>
   <td style="text-align:left;"> ENSG00000130489 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 25.2432774 </td>
   <td style="text-align:right;"> 54.6746318 </td>
   <td style="text-align:right;"> 32.5065443 </td>
   <td style="text-align:right;"> 24.4184588 </td>
   <td style="text-align:right;"> 6.1012456 </td>
   <td style="text-align:right;"> 32.7093975 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 3938 </td>
   <td style="text-align:left;"> ENSG00000130723 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 82.7061237 </td>
   <td style="text-align:right;"> 95.9575296 </td>
   <td style="text-align:right;"> 78.4229905 </td>
   <td style="text-align:right;"> 88.7995465 </td>
   <td style="text-align:right;"> 52.0211467 </td>
   <td style="text-align:right;"> 44.3204691 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 5959 </td>
   <td style="text-align:left;"> ENSG00000155640 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 13.6965176 </td>
   <td style="text-align:right;"> 16.8051716 </td>
   <td style="text-align:right;"> 13.8532045 </td>
   <td style="text-align:right;"> 20.1013832 </td>
   <td style="text-align:right;"> 3.8534183 </td>
   <td style="text-align:right;"> 4.9776974 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 8284 </td>
   <td style="text-align:left;"> ENSG00000179979 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 13.4240760 </td>
   <td style="text-align:right;"> 12.1304427 </td>
   <td style="text-align:right;"> 19.0536974 </td>
   <td style="text-align:right;"> 6.6105220 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 11.5790722 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 8418 </td>
   <td style="text-align:left;"> ENSG00000182109 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.1910015 </td>
   <td style="text-align:right;"> 3.9754169 </td>
   <td style="text-align:right;"> 1.5238525 </td>
   <td style="text-align:right;"> 6.1707199 </td>
   <td style="text-align:right;"> 1.2844728 </td>
   <td style="text-align:right;"> 3.3184649 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 8936 </td>
   <td style="text-align:left;"> ENSG00000188206 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 8.0869004 </td>
   <td style="text-align:right;"> 15.9197378 </td>
   <td style="text-align:right;"> 7.9240330 </td>
   <td style="text-align:right;"> 11.9002887 </td>
   <td style="text-align:right;"> 11.9231184 </td>
   <td style="text-align:right;"> 23.8253928 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9538 </td>
   <td style="text-align:left;"> ENSG00000203812 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.9408912 </td>
   <td style="text-align:right;"> 8.7386892 </td>
   <td style="text-align:right;"> 2.9562738 </td>
   <td style="text-align:right;"> 6.4958497 </td>
   <td style="text-align:right;"> 9.6431792 </td>
   <td style="text-align:right;"> 4.7359235 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9580 </td>
   <td style="text-align:left;"> ENSG00000204282 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 24.0760959 </td>
   <td style="text-align:right;"> 13.4784704 </td>
   <td style="text-align:right;"> 10.2305915 </td>
   <td style="text-align:right;"> 18.7253153 </td>
   <td style="text-align:right;"> 14.8870393 </td>
   <td style="text-align:right;"> 9.5085871 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9824 </td>
   <td style="text-align:left;"> ENSG00000213865 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.7622410 </td>
   <td style="text-align:right;"> 1.3968169 </td>
   <td style="text-align:right;"> 2.8773106 </td>
   <td style="text-align:right;"> 2.7966555 </td>
   <td style="text-align:right;"> 4.3543626 </td>
   <td style="text-align:right;"> 0.3887345 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9838 </td>
   <td style="text-align:left;"> ENSG00000213971 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 18.1791496 </td>
   <td style="text-align:right;"> 11.5919544 </td>
   <td style="text-align:right;"> 4.8915665 </td>
   <td style="text-align:right;"> 7.2783196 </td>
   <td style="text-align:right;"> 11.8685283 </td>
   <td style="text-align:right;"> 8.9752624 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9919 </td>
   <td style="text-align:left;"> ENSG00000215271 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.1910015 </td>
   <td style="text-align:right;"> 4.3368185 </td>
   <td style="text-align:right;"> 3.4369800 </td>
   <td style="text-align:right;"> 6.0695384 </td>
   <td style="text-align:right;"> 0.6261805 </td>
   <td style="text-align:right;"> 3.6965329 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9974 </td>
   <td style="text-align:left;"> ENSG00000221995 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 16.7603690 </td>
   <td style="text-align:right;"> 13.5597857 </td>
   <td style="text-align:right;"> 3.0463197 </td>
   <td style="text-align:right;"> 2.1855195 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.9339108 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 9978 </td>
   <td style="text-align:left;"> ENSG00000223473 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.2331279 </td>
   <td style="text-align:right;"> 1.8070077 </td>
   <td style="text-align:right;"> 2.9091729 </td>
   <td style="text-align:right;"> 0.4047258 </td>
   <td style="text-align:right;"> 0.9633546 </td>
   <td style="text-align:right;"> 2.9629151 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10025 </td>
   <td style="text-align:left;"> ENSG00000224959 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.7443760 </td>
   <td style="text-align:right;"> 3.2526138 </td>
   <td style="text-align:right;"> 6.3724741 </td>
   <td style="text-align:right;"> 10.5255699 </td>
   <td style="text-align:right;"> 1.6055909 </td>
   <td style="text-align:right;"> 3.7925313 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10047 </td>
   <td style="text-align:left;"> ENSG00000225860 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.8441223 </td>
   <td style="text-align:right;"> 1.0661345 </td>
   <td style="text-align:right;"> 0.6912749 </td>
   <td style="text-align:right;"> 1.2586973 </td>
   <td style="text-align:right;"> 0.4110313 </td>
   <td style="text-align:right;"> 1.3617558 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10067 </td>
   <td style="text-align:left;"> ENSG00000226380 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 54.6952565 </td>
   <td style="text-align:right;"> 22.2261946 </td>
   <td style="text-align:right;"> 19.8876603 </td>
   <td style="text-align:right;"> 16.3860000 </td>
   <td style="text-align:right;"> 41.4242464 </td>
   <td style="text-align:right;"> 6.2102700 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10103 </td>
   <td style="text-align:left;"> ENSG00000228106 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.0421263 </td>
   <td style="text-align:right;"> 5.7824246 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 1.4839947 </td>
   <td style="text-align:right;"> 1.9267091 </td>
   <td style="text-align:right;"> 4.9776974 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10117 </td>
   <td style="text-align:left;"> ENSG00000228439 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.1910015 </td>
   <td style="text-align:right;"> 1.2649054 </td>
   <td style="text-align:right;"> 0.9697243 </td>
   <td style="text-align:right;"> 3.2378067 </td>
   <td style="text-align:right;"> 0.9633546 </td>
   <td style="text-align:right;"> 1.6592325 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10145 </td>
   <td style="text-align:left;"> ENSG00000229729 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 9.5741635 </td>
   <td style="text-align:right;"> 18.0736910 </td>
   <td style="text-align:right;"> 11.4981597 </td>
   <td style="text-align:right;"> 19.0221143 </td>
   <td style="text-align:right;"> 22.4397391 </td>
   <td style="text-align:right;"> 10.7660483 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10214 </td>
   <td style="text-align:left;"> ENSG00000232559 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 7.5673260 </td>
   <td style="text-align:right;"> 6.9949268 </td>
   <td style="text-align:right;"> 7.0374279 </td>
   <td style="text-align:right;"> 7.8273977 </td>
   <td style="text-align:right;"> 8.0279547 </td>
   <td style="text-align:right;"> 4.8153296 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10257 </td>
   <td style="text-align:left;"> ENSG00000233864 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 49.8731890 </td>
   <td style="text-align:right;"> 38.8506654 </td>
   <td style="text-align:right;"> 56.9352850 </td>
   <td style="text-align:right;"> 37.9093199 </td>
   <td style="text-align:right;"> 39.1764191 </td>
   <td style="text-align:right;"> 39.4672142 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10269 </td>
   <td style="text-align:left;"> ENSG00000234290 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 25.4844552 </td>
   <td style="text-align:right;"> 52.6110290 </td>
   <td style="text-align:right;"> 7.1205471 </td>
   <td style="text-align:right;"> 3.0840109 </td>
   <td style="text-align:right;"> 38.8553009 </td>
   <td style="text-align:right;"> 8.5995648 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10273 </td>
   <td style="text-align:left;"> ENSG00000234449 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.2977504 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.2881467 </td>
   <td style="text-align:right;"> 0.2738645 </td>
   <td style="text-align:right;"> 0.8252737 </td>
   <td style="text-align:right;"> 0.1185166 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10338 </td>
   <td style="text-align:left;"> ENSG00000236673 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.1518527 </td>
   <td style="text-align:right;"> 1.7220783 </td>
   <td style="text-align:right;"> 2.8399069 </td>
   <td style="text-align:right;"> 2.3271736 </td>
   <td style="text-align:right;"> 2.5528896 </td>
   <td style="text-align:right;"> 1.9069322 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10346 </td>
   <td style="text-align:left;"> ENSG00000236850 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 8.5588347 </td>
   <td style="text-align:right;"> 10.5366619 </td>
   <td style="text-align:right;"> 8.0099228 </td>
   <td style="text-align:right;"> 2.4796203 </td>
   <td style="text-align:right;"> 4.8488847 </td>
   <td style="text-align:right;"> 14.4021377 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10366 </td>
   <td style="text-align:left;"> ENSG00000237513 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 4.4662557 </td>
   <td style="text-align:right;"> 1.8070077 </td>
   <td style="text-align:right;"> 2.9091729 </td>
   <td style="text-align:right;"> 8.2294253 </td>
   <td style="text-align:right;"> 10.5969002 </td>
   <td style="text-align:right;"> 6.5184132 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10407 </td>
   <td style="text-align:left;"> ENSG00000239665 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 13.6607875 </td>
   <td style="text-align:right;"> 4.8789208 </td>
   <td style="text-align:right;"> 0.6926602 </td>
   <td style="text-align:right;"> 17.8592020 </td>
   <td style="text-align:right;"> 16.4701519 </td>
   <td style="text-align:right;"> 8.0152779 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10473 </td>
   <td style="text-align:left;"> ENSG00000241978 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 20.8187067 </td>
   <td style="text-align:right;"> 18.7928800 </td>
   <td style="text-align:right;"> 14.8229288 </td>
   <td style="text-align:right;"> 30.0846205 </td>
   <td style="text-align:right;"> 36.9285918 </td>
   <td style="text-align:right;"> 28.4439849 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10492 </td>
   <td style="text-align:left;"> ENSG00000242687 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 9.0843642 </td>
   <td style="text-align:right;"> 2.3183909 </td>
   <td style="text-align:right;"> 5.4429240 </td>
   <td style="text-align:right;"> 11.6965767 </td>
   <td style="text-align:right;"> 13.8080821 </td>
   <td style="text-align:right;"> 2.2518155 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10503 </td>
   <td style="text-align:left;"> ENSG00000243251 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.5899870 </td>
   <td style="text-align:right;"> 1.2161162 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 2.6550015 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10638 </td>
   <td style="text-align:left;"> ENSG00000248632 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 8.5186384 </td>
   <td style="text-align:right;"> 6.1402121 </td>
   <td style="text-align:right;"> 6.4639052 </td>
   <td style="text-align:right;"> 6.3083267 </td>
   <td style="text-align:right;"> 13.1658458 </td>
   <td style="text-align:right;"> 1.3072381 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10735 </td>
   <td style="text-align:left;"> ENSG00000254561 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.5701920 </td>
   <td style="text-align:right;"> 1.0787836 </td>
   <td style="text-align:right;"> 1.2481737 </td>
   <td style="text-align:right;"> 3.0745673 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 1.6236775 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10785 </td>
   <td style="text-align:left;"> ENSG00000256045 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.3751655 </td>
   <td style="text-align:right;"> 1.0787836 </td>
   <td style="text-align:right;"> 0.9794216 </td>
   <td style="text-align:right;"> 1.4812966 </td>
   <td style="text-align:right;"> 1.3165846 </td>
   <td style="text-align:right;"> 1.8690068 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10792 </td>
   <td style="text-align:left;"> ENSG00000256222 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 10.1949731 </td>
   <td style="text-align:right;"> 9.7271224 </td>
   <td style="text-align:right;"> 8.8868307 </td>
   <td style="text-align:right;"> 7.2877632 </td>
   <td style="text-align:right;"> 8.9591975 </td>
   <td style="text-align:right;"> 8.0721659 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10806 </td>
   <td style="text-align:left;"> ENSG00000256618 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 5.4890283 </td>
   <td style="text-align:right;"> 5.5167945 </td>
   <td style="text-align:right;"> 2.6418061 </td>
   <td style="text-align:right;"> 4.4519842 </td>
   <td style="text-align:right;"> 3.8534183 </td>
   <td style="text-align:right;"> 5.7610921 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10819 </td>
   <td style="text-align:left;"> ENSG00000257151 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.8932511 </td>
   <td style="text-align:right;"> 0.3614015 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 1.2141775 </td>
   <td style="text-align:right;"> 3.8534183 </td>
   <td style="text-align:right;"> 8.8887453 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10854 </td>
   <td style="text-align:left;"> ENSG00000258297 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 16.9166880 </td>
   <td style="text-align:right;"> 34.0584810 </td>
   <td style="text-align:right;"> 42.7662275 </td>
   <td style="text-align:right;"> 26.3503501 </td>
   <td style="text-align:right;"> 34.7192986 </td>
   <td style="text-align:right;"> 43.9151423 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10880 </td>
   <td style="text-align:left;"> ENSG00000258838 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 16.9390192 </td>
   <td style="text-align:right;"> 13.2544014 </td>
   <td style="text-align:right;"> 15.8619191 </td>
   <td style="text-align:right;"> 3.6668161 </td>
   <td style="text-align:right;"> 8.0664889 </td>
   <td style="text-align:right;"> 9.9779129 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10919 </td>
   <td style="text-align:left;"> ENSG00000259834 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 3.8707550 </td>
   <td style="text-align:right;"> 7.4087315 </td>
   <td style="text-align:right;"> 14.1302686 </td>
   <td style="text-align:right;"> 16.0541248 </td>
   <td style="text-align:right;"> 17.3403822 </td>
   <td style="text-align:right;"> 11.0232293 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10952 </td>
   <td style="text-align:left;"> ENSG00000260276 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 4.1416616 </td>
   <td style="text-align:right;"> 2.5254392 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.3211182 </td>
   <td style="text-align:right;"> 2.9783223 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10974 </td>
   <td style="text-align:left;"> ENSG00000260822 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 6.6993836 </td>
   <td style="text-align:right;"> 10.8420462 </td>
   <td style="text-align:right;"> 18.7018260 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 18.3037368 </td>
   <td style="text-align:right;"> 1.7777491 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 10993 </td>
   <td style="text-align:left;"> ENSG00000261254 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.8381673 </td>
   <td style="text-align:right;"> 13.0140694 </td>
   <td style="text-align:right;"> 9.4229497 </td>
   <td style="text-align:right;"> 7.0152478 </td>
   <td style="text-align:right;"> 0.2761616 </td>
   <td style="text-align:right;"> 2.1036697 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11004 </td>
   <td style="text-align:left;"> ENSG00000261490 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.6376271 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 9.1431149 </td>
   <td style="text-align:right;"> 2.9679895 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 4.4680760 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11008 </td>
   <td style="text-align:left;"> ENSG00000261533 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.0391488 </td>
   <td style="text-align:right;"> 1.8756740 </td>
   <td style="text-align:right;"> 0.2867613 </td>
   <td style="text-align:right;"> 2.0128365 </td>
   <td style="text-align:right;"> 0.9633546 </td>
   <td style="text-align:right;"> 1.0702049 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11042 </td>
   <td style="text-align:left;"> ENSG00000262621 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 4.8726850 </td>
   <td style="text-align:right;"> 9.6060529 </td>
   <td style="text-align:right;"> 16.9549369 </td>
   <td style="text-align:right;"> 5.0239967 </td>
   <td style="text-align:right;"> 19.3216815 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11172 </td>
   <td style="text-align:left;"> ENSG00000267654 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.0421263 </td>
   <td style="text-align:right;"> 1.0842046 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 3.3727153 </td>
   <td style="text-align:right;"> 5.1378910 </td>
   <td style="text-align:right;"> 1.1851660 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11191 </td>
   <td style="text-align:left;"> ENSG00000268220 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 20.5164901 </td>
   <td style="text-align:right;"> 33.4621685 </td>
   <td style="text-align:right;"> 27.0566936 </td>
   <td style="text-align:right;"> 6.5606058 </td>
   <td style="text-align:right;"> 16.9614628 </td>
   <td style="text-align:right;"> 15.2388649 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11195 </td>
   <td style="text-align:left;"> ENSG00000268439 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 6.3986557 </td>
   <td style="text-align:right;"> 20.9793593 </td>
   <td style="text-align:right;"> 16.9535516 </td>
   <td style="text-align:right;"> 8.1592728 </td>
   <td style="text-align:right;"> 17.2247797 </td>
   <td style="text-align:right;"> 18.3961472 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11201 </td>
   <td style="text-align:left;"> ENSG00000268861 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 11.4038396 </td>
   <td style="text-align:right;"> 4.0476972 </td>
   <td style="text-align:right;"> 7.0069508 </td>
   <td style="text-align:right;"> 11.0071936 </td>
   <td style="text-align:right;"> 3.1951260 </td>
   <td style="text-align:right;"> 7.8078739 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11247 </td>
   <td style="text-align:left;"> ENSG00000270195 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 3.7218798 </td>
   <td style="text-align:right;"> 3.9754169 </td>
   <td style="text-align:right;"> 3.3247691 </td>
   <td style="text-align:right;"> 2.9679895 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11260 </td>
   <td style="text-align:left;"> ENSG00000270672 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.1969565 </td>
   <td style="text-align:right;"> 1.8232708 </td>
   <td style="text-align:right;"> 1.1082564 </td>
   <td style="text-align:right;"> 1.1804504 </td>
   <td style="text-align:right;"> 0.3211182 </td>
   <td style="text-align:right;"> 1.9211541 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11274 </td>
   <td style="text-align:left;"> ENSG00000271430 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 10.0193004 </td>
   <td style="text-align:right;"> 31.1744967 </td>
   <td style="text-align:right;"> 2.3314943 </td>
   <td style="text-align:right;"> 13.6257698 </td>
   <td style="text-align:right;"> 2.1065353 </td>
   <td style="text-align:right;"> 10.4792381 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11284 </td>
   <td style="text-align:left;"> ENSG00000271840 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 2.5298108 </td>
   <td style="text-align:right;"> 0.1385320 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 7.7068365 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11288 </td>
   <td style="text-align:left;"> ENSG00000271895 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 12.9521416 </td>
   <td style="text-align:right;"> 3.2634559 </td>
   <td style="text-align:right;"> 11.6366918 </td>
   <td style="text-align:right;"> 14.3192001 </td>
   <td style="text-align:right;"> 12.4914976 </td>
   <td style="text-align:right;"> 11.5707760 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11294 </td>
   <td style="text-align:left;"> ENSG00000272053 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 5.8344187 </td>
   <td style="text-align:right;"> 7.0834702 </td>
   <td style="text-align:right;"> 8.8244912 </td>
   <td style="text-align:right;"> 3.1298798 </td>
   <td style="text-align:right;"> 11.5570436 </td>
   <td style="text-align:right;"> 12.3683928 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11300 </td>
   <td style="text-align:left;"> ENSG00000272196 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 5.8433513 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 8.6928858 </td>
   <td style="text-align:right;"> 3.2944683 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11323 </td>
   <td style="text-align:left;"> ENSG00000272579 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 14.7297114 </td>
   <td style="text-align:right;"> 15.3595654 </td>
   <td style="text-align:right;"> 26.6106205 </td>
   <td style="text-align:right;"> 18.9249801 </td>
   <td style="text-align:right;"> 26.7138222 </td>
   <td style="text-align:right;"> 17.1647597 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11344 </td>
   <td style="text-align:left;"> ENSG00000272869 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.1563190 </td>
   <td style="text-align:right;"> 2.6020911 </td>
   <td style="text-align:right;"> 0.3989723 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.7835284 </td>
   <td style="text-align:right;"> 9.6887324 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11354 </td>
   <td style="text-align:left;"> ENSG00000273000 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 2.8894053 </td>
   <td style="text-align:right;"> 9.2664085 </td>
   <td style="text-align:right;"> 0.4789256 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.0000000 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11410 </td>
   <td style="text-align:left;"> ENSG00000274627 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.6797534 </td>
   <td style="text-align:right;"> 16.0823685 </td>
   <td style="text-align:right;"> 5.8183459 </td>
   <td style="text-align:right;"> 9.4409047 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 14.2753249 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11413 </td>
   <td style="text-align:left;"> ENSG00000274956 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 4.9188363 </td>
   <td style="text-align:right;"> 6.3263339 </td>
   <td style="text-align:right;"> 4.6241997 </td>
   <td style="text-align:right;"> 3.6236453 </td>
   <td style="text-align:right;"> 4.4378534 </td>
   <td style="text-align:right;"> 4.7797746 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11473 </td>
   <td style="text-align:left;"> ENSG00000277203 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.4466256 </td>
   <td style="text-align:right;"> 3.0719131 </td>
   <td style="text-align:right;"> 0.5541282 </td>
   <td style="text-align:right;"> 3.7774411 </td>
   <td style="text-align:right;"> 1.2844728 </td>
   <td style="text-align:right;"> 3.0553580 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11521 </td>
   <td style="text-align:left;"> ENSG00000279007 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.8341424 </td>
   <td style="text-align:right;"> 1.3136946 </td>
   <td style="text-align:right;"> 0.6912749 </td>
   <td style="text-align:right;"> 1.5824780 </td>
   <td style="text-align:right;"> 2.4501318 </td>
   <td style="text-align:right;"> 1.3309415 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11522 </td>
   <td style="text-align:left;"> ENSG00000279010 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 23.5416340 </td>
   <td style="text-align:right;"> 27.5351832 </td>
   <td style="text-align:right;"> 34.6496350 </td>
   <td style="text-align:right;"> 25.2036269 </td>
   <td style="text-align:right;"> 25.5899085 </td>
   <td style="text-align:right;"> 23.8005044 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11546 </td>
   <td style="text-align:left;"> ENSG00000279184 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 58.6895779 </td>
   <td style="text-align:right;"> 84.1866815 </td>
   <td style="text-align:right;"> 64.4908228 </td>
   <td style="text-align:right;"> 82.4777290 </td>
   <td style="text-align:right;"> 52.3037307 </td>
   <td style="text-align:right;"> 72.4254965 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11547 </td>
   <td style="text-align:left;"> ENSG00000279191 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.7443760 </td>
   <td style="text-align:right;"> 1.0787836 </td>
   <td style="text-align:right;"> 0.5541282 </td>
   <td style="text-align:right;"> 5.8010703 </td>
   <td style="text-align:right;"> 1.9748769 </td>
   <td style="text-align:right;"> 0.4693258 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11564 </td>
   <td style="text-align:left;"> ENSG00000279380 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.5308782 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 2.9091729 </td>
   <td style="text-align:right;"> 2.6981722 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 3.4369815 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11566 </td>
   <td style="text-align:left;"> ENSG00000279396 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.3820031 </td>
   <td style="text-align:right;"> 1.0842046 </td>
   <td style="text-align:right;"> 5.5094194 </td>
   <td style="text-align:right;"> 0.8404807 </td>
   <td style="text-align:right;"> 4.7653939 </td>
   <td style="text-align:right;"> 3.0743207 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11609 </td>
   <td style="text-align:left;"> ENSG00000279777 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.3056354 </td>
   <td style="text-align:right;"> 3.3935604 </td>
   <td style="text-align:right;"> 2.0752100 </td>
   <td style="text-align:right;"> 3.3362900 </td>
   <td style="text-align:right;"> 0.4945220 </td>
   <td style="text-align:right;"> 1.1389446 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11624 </td>
   <td style="text-align:left;"> ENSG00000279921 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 3.0310989 </td>
   <td style="text-align:right;"> 1.5775177 </td>
   <td style="text-align:right;"> 1.6388341 </td>
   <td style="text-align:right;"> 3.5130203 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 1.1211671 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11628 </td>
   <td style="text-align:left;"> ENSG00000279948 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 16.2288846 </td>
   <td style="text-align:right;"> 14.8174631 </td>
   <td style="text-align:right;"> 4.3485209 </td>
   <td style="text-align:right;"> 4.8836918 </td>
   <td style="text-align:right;"> 2.3120510 </td>
   <td style="text-align:right;"> 0.2370332 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11631 </td>
   <td style="text-align:left;"> ENSG00000280006 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.6350909 </td>
   <td style="text-align:right;"> 3.6989447 </td>
   <td style="text-align:right;"> 1.3950177 </td>
   <td style="text-align:right;"> 3.8988589 </td>
   <td style="text-align:right;"> 2.9221755 </td>
   <td style="text-align:right;"> 2.1427802 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11646 </td>
   <td style="text-align:left;"> ENSG00000280130 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 2.7020847 </td>
   <td style="text-align:right;"> 2.0039715 </td>
   <td style="text-align:right;"> 3.0186133 </td>
   <td style="text-align:right;"> 3.1325780 </td>
   <td style="text-align:right;"> 4.1970147 </td>
   <td style="text-align:right;"> 3.4026117 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11651 </td>
   <td style="text-align:left;"> ENSG00000280156 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 107.1663176 </td>
   <td style="text-align:right;"> 162.5656402 </td>
   <td style="text-align:right;"> 129.3695353 </td>
   <td style="text-align:right;"> 148.0109363 </td>
   <td style="text-align:right;"> 90.3273355 </td>
   <td style="text-align:right;"> 124.9129448 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11652 </td>
   <td style="text-align:left;"> ENSG00000280158 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 3.6921047 </td>
   <td style="text-align:right;"> 7.7231509 </td>
   <td style="text-align:right;"> 1.5058433 </td>
   <td style="text-align:right;"> 1.8684843 </td>
   <td style="text-align:right;"> 3.0506228 </td>
   <td style="text-align:right;"> 6.1806409 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11666 </td>
   <td style="text-align:left;"> ENSG00000280328 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 1.4664206 </td>
   <td style="text-align:right;"> 1.9913225 </td>
   <td style="text-align:right;"> 3.1460627 </td>
   <td style="text-align:right;"> 2.7494375 </td>
   <td style="text-align:right;"> 3.2721943 </td>
   <td style="text-align:right;"> 2.4094426 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11679 </td>
   <td style="text-align:left;"> ENSG00000280447 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 3.1323340 </td>
   <td style="text-align:right;"> 1.9840944 </td>
   <td style="text-align:right;"> 2.5102006 </td>
   <td style="text-align:right;"> 4.9848732 </td>
   <td style="text-align:right;"> 4.9741208 </td>
   <td style="text-align:right;"> 2.8894348 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11707 </td>
   <td style="text-align:left;"> ENSG00000281657 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 7.4080295 </td>
   <td style="text-align:right;"> 7.9165007 </td>
   <td style="text-align:right;"> 4.9871536 </td>
   <td style="text-align:right;"> 8.0054770 </td>
   <td style="text-align:right;"> 7.7871161 </td>
   <td style="text-align:right;"> 6.6594480 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11712 </td>
   <td style="text-align:left;"> ENSG00000282031 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 38.0867402 </td>
   <td style="text-align:right;"> 37.6815314 </td>
   <td style="text-align:right;"> 25.8916392 </td>
   <td style="text-align:right;"> 23.1179397 </td>
   <td style="text-align:right;"> 37.1983310 </td>
   <td style="text-align:right;"> 23.4662875 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11720 </td>
   <td style="text-align:left;"> ENSG00000282827 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 15.7263880 </td>
   <td style="text-align:right;"> 13.3655717 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 0.0963355 </td>
   <td style="text-align:right;"> 8.7358589 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11726 </td>
   <td style="text-align:left;"> ENSG00000283013 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 9.6768874 </td>
   <td style="text-align:right;"> 10.5023287 </td>
   <td style="text-align:right;"> 13.4957918 </td>
   <td style="text-align:right;"> 25.4977276 </td>
   <td style="text-align:right;"> 15.4136731 </td>
   <td style="text-align:right;"> 18.5928848 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 11729 </td>
   <td style="text-align:left;"> ENSG00000283088 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:right;"> 7.9648227 </td>
   <td style="text-align:right;"> 0.0000000 </td>
   <td style="text-align:right;"> 6.1148045 </td>
   <td style="text-align:right;"> 16.7502533 </td>
   <td style="text-align:right;"> 0.1958821 </td>
   <td style="text-align:right;"> 21.3898766 </td>
  </tr>
</tbody>
</table>

<!-- rnb-chunk-end -->


<!-- rnb-text-begin -->


Future development of such a study may involve investigating the ID mappings, how they changed and for what reasons.

## Additional Notes

-   Most `kable` displayed tables have been cropped due to relatively large amounts of data.
-   There does seem to be a discrepancy between provided supplementary materials and samples noted as part of the original paper.
-   "Why is the data set of interest to you?" at [Data set's Importance].
-   "What are the control and test conditions of the dataset?" at [Variables of Data Set].
-   "How many samples in each of the conditions of your dataset?" at [Overview of Data].
-   "Were there expression values that were not unique for specific genes? How did you handle these?" at [Checking for Identical Expression Values].
-   "Were there expression values that could not be mapped to current HUGO symbols?" at [(Re)Mapping HUGO Symbols].
-   "Were there any outliers in your dataset? How were they handled in the originating paper? How many outliers were removed?" at [Filtering the dataset].
-   "How did you handle replicates?" at [Fetching Sample Metadata].

# References

::: {#refs}
:::

<!-- rnb-text-end -->

