---
title: "Assignment 2"
author: "Harrison Deng"
date: "`r Sys.Date()`"
output: 
  html_notebook:
    toc: yes
bibliography: references.bib
---

```{r message=FALSE}
library("GEOquery")
library("knitr")
library("edgeR")
library("stringr")
library("limma")
library("ggplot2")
library("ComplexHeatmap")
library("gprofiler2")
library("circlize")
library("data.table")
```

Loading environment from assignment 1. We will use the `knitr` package to do so 
[@xieKnitrGeneralPurposePackage2023].
By doing so, we will inherently include all packages used in assignment 1. 
Refer to assignment 1 for references on those packages.

```{r message=FALSE}
if (basename(getwd()) == "Homework") {
  setwd("./Assignment 2")
}
preprocessed_workspace_path <- "preprocessed_workspace.RData"
if (exists("n_ctrl_vs_art")) {
  # Then continue
} else if (file.exists(preprocessed_workspace_path)) {
  print("Loaded from file cache.")
  load(preprocessed_workspace_path)
} else {
  part_a1_dir <- file.path(dirname(getwd()), "Assignment 1")
  part_a1_filename <- "a1_deng_yunyang.Rmd"
  part_a1_out <- "a1_deng_yunyang.nb.html"

  knit(file.path(part_a1_dir, part_a1_filename),
       output = file.path(getwd(), part_a1_out))
  save.image(preprocessed_workspace_path)
}
```

# Summary 


## Data set

The gene expression analysis will be performed on the normalized dataset 
generated in the previous assignment (#1). Aforementioned data set was procured 
from the Gene Expression Omnibus (GEO) with `r data_set_geoid`. The dataset 
contains a variety of categories to stratify upon and categorize for testing. 
The variables used in the previous assignment and the variables we will continue 
to use in this assignment are the phenotype categories which designates if the 
patient sample is from a patient who has undergone antiretroviral therapy (ART), 
is an elite controller with no signs of viremia (EC), or a viremic controller 
with very low signs of viremia (VC). More specifically, there were $6$ patient 
samples, $18$ and $24$ respectively.

## Data set Pre-processing

The dataset was pre-processed via filtering out genes based on patient sample 
counts where the sum of counts per million (CPM) over patient samples must have 
been greater than $30$.

Duplicate rows with exact same values were simply and trivially deleted from the
 dataset. This reduced the number of genes (rows) by $14$.

The same dataset was than normalized via trimmed mean of m-values (TMM). TMM was
used as we expect that most genes would not be differentially expressed. 
Furthermore, with TMM, we hoped to reduce technical variation as the process of 
recording gene expression is very complex with many steps that may introduce 
such variation.

## Preliminary Analysis

The resulting data was pre-emptively analysed at this stage via multidimensional 
scaling plots (MDS). The produced plot generated 3 groupings where one group was 
entirely VCs, another was primarily EC and VC, and the last was a mixture of EC, 
and ART patients.

A dispersion analysis via biological coefficient of variation (BCV) indicated 
that the produced pre-processed dataset shows that as gene expression increases, 
the biological coefficient of variation decreases approximately after 
$3\log CPM$. This result was validated by plotting the mean variance and seeing 
that the data largely follows the negative binomial distribution.

# Gene Expression Differential Analysis

> 1. Calculate p-values for each of the genes in your expression set. How many 
> genes were significantly differentially expressed? What thresholds did you use 
> and why?

Let's review the data we have currently from assignment 1 (filtered and 
normalized).

```{r}
kable(
  n_ctrl_vs_art[1:5, 1:4],
  type = "html",
  caption = "Sample of Pre-processed Gene Expression Data Format"
)

t(art_v_ec_meta[1, ])
```

Recall that samples are the columns and rows are the genes. Next, we will look 
at the MDS plots color coded by patients and phenotype. This seems to indicate 
the two patients 

```{r}

phenotype_dge_mds_by <- function(classification_var, title, dge) {
  class_factors <- factor(classification_var)
  col <- c(rainbow(length(unique(classification_var))))
  class_col <- col[class_factors]

  limma::plotMDS(dge, labels = NULL, pch = 1, col = class_col)

  legend("topright", legend = levels(class_factors),
         pch = c(1), col = col, title = title,
         bty = "n", cex = 0.75)

}

par(mfrow = c(2, 3))
phenotype_dge_mds_by(art_v_ec_meta$Patient, "Patient", phenotype_dge)
phenotype_dge_mds_by(art_v_ec_meta$Phenotype, "Phenotype", phenotype_dge)
phenotype_dge_mds_by(art_v_ec_meta$Source, "Phenotype", phenotype_dge)
phenotype_dge_mds_by(art_v_ec_meta$Sex, "Phenotype", phenotype_dge)
phenotype_dge_mds_by(art_v_ec_meta$Race, "Phenotype", phenotype_dge)
phenotype_dge_mds_by(art_v_ec_meta$Age, "Phenotype", phenotype_dge)
par(mfrow = c(1, 1))
```

Immediately, it is apparent that the sex, race, age and phenotype seem to have 
some influence on the genetic expression. To elaborate, the clustering when 
colouring based on sex and race show to very apparent groups. Furthermore, 
samples from similar patients are clearly similar Lastly, the clustering on the 
phenotype also seems rather distinct and forms interesting intersections with
each other. Since we are interested in the associations between gene expression 
and phenotypic outcome, that is to say, we are interested in how gene 
expressions may be linked to HIV severity outcome, we will elect to look at the 
phenotype category more. Additionally, since single patients have multiple 
samples that are also relatively clustered together, we'll look at those.

All figures above were generated via use of the `limma` package 
[@ritchieLimmaPowersDifferential2015].

> 2. Multiple hypothesis testing - correct your p-values using a multiple 
> hypothesis correction method. Which method did you use? And Why? How many 
> genes passed correction?

Throughout the gene expression analysis and specifically, the fitting and 
testing of the genewise negative binomial generalized linear models, `EdgeR`
functions will be used [@chenEdgeRPowerfulDifferential2024].

We will begin by creating design pattern matrix for the model and fitting the 
expression matrix to the design pattern.

```{r}
model_design_pat <- model.matrix(~ art_v_ec_meta$Patient + art_v_ec_meta$Phenotype)

model_design_pat[1:5,]

fit_phenotype_glmQL <- glmQLFit(phenotype_dge, model_design_pat)
```

Specifically, the reference category will be the ART samples while we will test 
the elite controllers.

```{r}
coefficent_col <- "art_v_ec_meta$PhenotypeEC"

qlf.ARTvsEC  <- glmQLFTest(fit_phenotype_glmQL, coef = coefficent_col)

qlf_art_vs_ec_results <- topTags(qlf.ARTvsEC, sort.by = "PValue",
                           n = nrow(n_ctrl_vs_art))$table
qlf_art_vs_ec_results$logFDR <- -log2(qlf_art_vs_ec_results$FDR)

kable(
  qlf_art_vs_ec_results[1:10,],
  row.names = TRUE,
  type = "html",
  caption = "Top 10 Significant Gene Expr. Change In Elite Controllers"
)
```

We will let $\alpha = 0.05$. Thus, the following demonstrates the number of 
samples with p-values below 0.05. The $\alpha = 0.05$ was selected as that means
there is a very little probability ($5\%$) chance that the association we see is 
completely randomly generated, i.e., false positive.

### Summary of Results

```{r}
alpha <- 0.05
length(which(qlf_art_vs_ec_results$PValue < alpha))
```

Furthermore, the following demonstrates the amount passing multiple tests 
correction as follows.

```{r}
length(which(qlf_art_vs_ec_results$FDR < alpha))
```

The built in false discovery rate (FDR) multiple testing correction of 
`glmQLFTest` was chosen as it offers less false negatives compared to Bonferonni
Corrections.

> 3. Show the amount of differentially expressed genes using an MA Plot or a 
> Volcano plot. Highlight genes of interest.

We will generate a volcano plot to visualize and further discuss our results.

```{r}
# Create a basic volcano plot

signif_expr <- 0.25

qlf_art_vs_ec_results$regulation <- apply(
  qlf_art_vs_ec_results[c("FDR", "logFC")],
  MARGIN = 1,
  FUN = function(v) {
    if (v["FDR"] < alpha) {
      if (v["logFC"] >= signif_expr) {
        return("Upregulated")
      } else if (v["logFC"] <= -signif_expr) {
        return("Downregulated")
      }
    }
    return("Insignificant Change")
})

ggplot(
       qlf_art_vs_ec_results,
       aes(x = logFC, y = logFDR, color = regulation)) +
  geom_point(alpha = 0.4) +
  scale_color_manual(
                     values = c("blue", "grey", "red")) +
  theme_bw() +
  xlab("Log2 Fold Change") +
  ylab("-Log10 Adjusted P-Value") +
  geom_vline(xintercept = c(-signif_expr, signif_expr), linetype = "dotted") +
  geom_hline(yintercept = -log2(alpha), linetype = "dotted") +
  ggtitle("Expression and Association Levels of Genes")
```

The top right indicate the upregulated genes and the top left indicate the 
downregulated genes. We see that there are many genes that were not significant
after FDR P-value adjustments.

> 4. Visualize your top hits using a heatmap. Do you conditions cluster  
> together? Explain why or why not.

Our next step will be to plot a heatmap. This heatmap will be a re-scaled 
heatmap of only the significantly associated genes after FDR.

```{r}

unique_patient <- unique(art_v_ec_meta$Patient)
unique_patientcolors <- rainbow(n = length(unique_patient))
names(unique_patientcolors) <- unique_patient

unique_phenotypes <- unique(art_v_ec_meta$Phenotype)
unique_phenotypescolors <- rainbow(n = length(unique_phenotypes))
names(unique_phenotypescolors) <- unique_phenotypes

ha_pat <- HeatmapAnnotation(df = data.frame(
  patient = art_v_ec_meta$Patient,
  phenotypes = art_v_ec_meta$Phenotype),
  col = list(
      patient = unique_patientcolors,
      phenotypes = unique_phenotypescolors
    ),
  show_legend = TRUE
)

top_hits <- row.names(qlf_art_vs_ec_results)[qlf_art_vs_ec_results$FDR < alpha]
gene_names <- row.names(n_ctrl_vs_art)
art_v_ec_sig_norm <- t(scale(t(n_ctrl_vs_art[gene_names %in% top_hits,])))
if (min(art_v_ec_sig_norm) == 0) {
  heatmap_col <- colorRamp2(c(0, max(art_v_ec_sig_norm)), c("white", "red"))
} else {
  heatmap_col <- colorRamp2(c(min(art_v_ec_sig_norm), 0,
  max(art_v_ec_sig_norm)),
  c("blue", "white", "red"))
}

current_heatmap <- Heatmap(as.matrix(art_v_ec_sig_norm),
  top_annotation = ha_pat,
  cluster_rows = TRUE,
  cluster_columns = TRUE,
  show_row_dend = TRUE,
  show_column_dend = TRUE,
  col = heatmap_col,
  show_column_names = FALSE,
  show_row_names = FALSE,
  show_heatmap_legend = TRUE,
  column_title = ("Top hits EC vs ART")
)

current_heatmap
```


The above heatmap demonstrates how the different genes perform in association 
with different HIV outcomes. Immediately, the figure depicts very obvious 
clustering of genetic expression level changes in the column of patient $310817$ 
and antiretroviral therapy patients patients (ARTs). Specifically, we can see 
very significant clustering of upregulation occurring in genes placed in the 
upper half of the square, while genes in the lower have are very significantly 
downregulated. Generally speaking however, there is significant downregulation 
in the same set of genes that was previously described as upregulated in viremic 
controller patients (VCs). Elite controllers (ECs) seem to also have a tad more 
downregulated genes in the upper half set of genes however, still closer to the 
previously mentioned $310817$ ART than the general VC set of patients and genes. 
This corroborates the results seen by the MDS plots where we have some 
intersections between the different phenotypes.

This heatmap was generated with use of the library `ComplexHeatmap` 
[@guComplexHeatmapVisualization2022].

> Make sure all your figures have proper heading and labels. Every figure  
> included in the report should have a detailed figure legend.

# Threshold Overrepresentation Analysis

This section will answer the following questions as described by the outline for
assignment 2. Please see the end of this section for the answer to these 
questions.

> 1. Which method did you choose and why?

```{r}


#where to put all the generated files
working_dir <- "./g_profiler/generated_data"
if (!dir.exists(working_dir)) {
  dir.create(working_dir, recursive = TRUE)
}


# where to find the data files needed to run the analysis
data_dir <-  "./g_profiler/data"
if (!dir.exists(data_dir)) {
  dir.create(data_dir, recursive = TRUE)
}


# default min size of the genesets for example -  3
min_gs_size <- 3

#min intersection between your genelist and the geneset - for example 3
min_intersection <- 3

# organism parameter used for g:profiler.  
# First letter of first word in species name followed by 
# the second word for example - hsapiens
organism <- "hsapiens"

sources <- c("REAC", "WP", "GO:BP")
max_term_size <- 250

```


I chose to use a list of upregulated and downregulated genes that was restricted 
based off the previously used `r alpha` values as thresholds. Thresholding was 
selected due the size of the dataset as well as well as the basis in terms of 
statistics. The upregulation and downregulation were also previously determined 
by the the `r signif_expr` variable. The gene lists produced as a product of 
such thresholds and previous gene expression analysis was used as a query set 
for [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) programmatically and the 
results were returned and further filtered by setting an upper bound on term 
size by `r max_term_size`. P-values are adjusted for multiple testing via FDR.
Parameters can be seen below. 

IDs will be used from the given dataset and parsed with help from the `stringr` 
library [@wickhamStringrSimpleConsistent2023a]. 

```{r}
qlf_art_vs_ec_results$hgnc_symbol <- lapply(row.names(qlf_art_vs_ec_results), FUN = function(x) {
  return(unlist(str_split(x, "_"))[[2]])
})

regulated_genes_dir <- "regulated_genes"

if (!dir.exists("regulated_genes")) {
  dir.create(regulated_genes_dir)
}

upregulated_genes <- qlf_art_vs_ec_results$hgnc_symbol[
  which(qlf_art_vs_ec_results$PValue < 0.05
  & qlf_art_vs_ec_results$logFC > 0)]
downregulated_genes <- qlf_art_vs_ec_results$hgnc_symbol[
  which(qlf_art_vs_ec_results$PValue < 0.05
  & qlf_art_vs_ec_results$logFC < 0)]
write.table(x=upregulated_genes,
  file=file.path(getwd(), file.path(
    regulated_genes_dir,
    "upregulated_genes.txt")),
  sep = "\n",row.names = FALSE,
  col.names = FALSE,quote = FALSE)
write.table(x=downregulated_genes,
  file=file.path(getwd(), file.path(
    regulated_genes_dir,
    "downregulated_genes.txt")),
sep = "\n",row.names = FALSE,
col.names = FALSE,quote = FALSE)
```

The next steps involve running g:Profiler programmatically via R. We will try to 
cache the results when possible with use of the `data.table` library 
[@barrettDataTableExtension2024].

```{r}

query_gprofiler <- function(
    genes,
    cache_name = "") {

  cache_path <- file.path(working_dir, "results", cache_name)
  if (!dir.exists(dirname(cache_path)) && cache_name != "") {
    dir.create(dirname(cache_path))
  }
  if (file.exists(cache_path) && cache_name != "") {
    enrichment_results <- fread(cache_path)
  } else {
gp_results <- gost(query = unlist(genes),
                       significant = FALSE,
                       ordered_query = FALSE,
                       exclude_iea = TRUE,
                       evcodes = TRUE,
                       correction_method = "fdr",
                       organism = organism,
                       domain_scope = "annotated",
                       source = sources)
    enrichment_results <- gp_results$result
    if (cache_name != "") {
      fwrite(enrichment_results, file = cache_path)
    }
  }

  return(enrichment_results)
}

upregulated_enrichment <- query_gprofiler(
  upregulated_genes, 
  "upregulation_results.csv")

downregulated_enrichment <- query_gprofiler(
  downregulated_genes, 
  "downregulation_results.csv")

all_enrichment <- query_gprofiler(
  c(upregulated_genes, downregulated_genes), 
  "all_results.csv")

```

Briefly, configure to only show specific, relevent columns and maximum number
of results.

```{r}

important_columns <- c(
  "term_name",
  "p_value",
  "term_size",
  "intersection_size",
  "source"
)

number_of_results <- 15

```

> 2. What annotation data did you use and why? What version of the annotation 
> are you using?

For this assignment, I used `r sources` as described outlined in the 
tutorial. I used the latest version of the annotations which can be see in the
following code block.

```{r}
str(get_version_info(organism = organism))
```

> 3. How many genesets were returned with what thresholds?

All term sizes were capped at `r max_term_size` and adjusted p-values were 
capped at `r alpha`. The max term size was chosen to reduce highly 
broad-covering pathways. The alpha value choice was previously described.

Upregulated significant genes enrichment analysis as follows.

```{r}
thresholded_up <- upregulated_enrichment[
  upregulated_enrichment$term_size <= max_term_size & 
  upregulated_enrichment$p_value < alpha,]
ordered_thresholded_up <- thresholded_up[
  order(thresholded_up$intersection_size),]
kable(
  thresholded_up[1:number_of_results, ..important_columns],
  type = "html",
  caption = paste(
    "Top",
    number_of_results,
    "Pathways Enriched with Up-regulated Genes"
  )
)
```


Downregulated significant genes enrichment analysis as follows.

```{r}
thresholded_down <- downregulated_enrichment[
  downregulated_enrichment$term_size <= max_term_size & 
  downregulated_enrichment$p_value < alpha,]
ordered_thresholded_down <- thresholded_down[
  order(thresholded_down$intersection_size),]
kable(
  thresholded_down[1:number_of_results, ..important_columns],
  type = "html",
  caption = paste(
    "Top",
    number_of_results,
    "Pathways Enriched with Down-regulated Genes"
  )
)
```

All significant genes enrichment analysis as follows.

```{r}
thresholded_all <- all_enrichment[
  all_enrichment$term_size <= max_term_size & 
  all_enrichment$p_value < alpha,]
ordered_thresholded_all <- thresholded_all[
  order(thresholded_all$intersection_size),]

kable(
  thresholded_all[1:number_of_results, ..important_columns],
  type = "html",
  caption = paste(
    "Top",
    number_of_results,
    "Pathways Enriched with All Regulated Genes"
  )
)
```


> 4. Run the analysis using the up-regulated set of genes, and the 
> down-regulated set of genes separately. How do these results compare to using 
> the whole list (i.e all differentially expressed genes together vs. the 
> up-regulated and down regulated differentially expressed genes separately)?

In the union set of both up and down regulated genes, we see that there are many
more results. This is due to there being more potential genes to partake in 
various pathways than any individual. Furthermore, for the same reason, there
are many more results where the range of possible intersection sizes is 
increased and the relative number of larger term sizes may also be increased.


> Additional Notes: Present your results with the use of tables and screenshots. 
> All figures should have appropriate figure legends.
>
> If using figures create a figures directory in your repo and make sure all 
> references to the figures are relative in your Rmarkdown notebook.

# Interpretation

The assignment outline requires the following.

> 1. Do the over-representation results support conclusions or mechanism 
> discussed in the original paper?

Yes, the over-representation results support the general conclusions discussed 
in the original paper. To elaborate, the paper specifically discusses the 
increase of CD8+ T cells within germinal centers 
[@collinsCytolyticCD8Cells2023]. This corresponds with the several up-regulation 
of positive regulation pathways of T-cells. This ranges from T-cell 
differentiation pathways to T-cell activation pathways. 

> 2. Can you find evidence, i.e. publications, to support some of the results 
that you see. How does this evidence support your results.

Diving into the specifics, we can see there is an upregulation of various immune
cell differentiation. This corroborates the general theory that the virus has 
some capacity to surpress the operations of the immune 
system [@dengSuppressionNFkBActivity2018].

In terms of down-regulation, several 
pathways associated with the lifecycle of a cell have very large intersections 
and highly significant p-values. This is also a known effect of HIV-1 and has 
been previously studied [@piekna-przybylskaCD4MemoryCells2018]. Furthermore, the 
results obtained seem to indicate various regulations in mRNA machinery are 
downregulated. This is also a well known effect with various papers citing as 
such [@toro-ascuyInteractionsHIV1Unspliced2016; @wangRoleRNAModification2022].

# Additional Notes

- Throughout this notebook, `knitr` will be used for the displaying of tables 
[@xieKnitrGeneralPurposePackage2023].
- Throughout this notebook, `ggplot2` will be used for the generation of figures 
[@wickhamGgplot2ElegantGraphics2016].
- Coloring associated functions were largely obtained and used from the `circlize` 
library [@guCirclizeImplementsEnhances2014].


# References

::: {#refs}
:::
